{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8af24c60",
   "metadata": {},
   "source": [
    "## Interval variant count\n",
    "\n",
    "We would like to aggregate the data according to the intervals, to cehck how many variants there are in each interval <br>\n",
    "The following notebook shows 3 kind of interval analysis:\n",
    "1. by GH associated DSD genes\n",
    "2. by DSD genes which are 1mb from an interval\n",
    "3. by genes which are 1.5mb from an interval\n",
    "\n",
    "In each analysis, 3 subgroups will be analysed - \n",
    "1. all samples\n",
    "2. samples diagnosed with DSD\n",
    "\n",
    "Ken's data and Sinclair data will be analysed seperately"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "13a09e1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import sys\n",
    "import os\n",
    "sys.path.append(os.path.abspath(\"code\"))\n",
    "# these are  needed-\n",
    "import pandas as pd\n",
    "from GnomAD_df_class import GnomAD_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "52fd4f01",
   "metadata": {},
   "outputs": [],
   "source": [
    "gdf = GnomAD_df(path='data/main_new_AF.prq',\n",
    "                peak_file='data/hATAC_mATAC_peak.bed.gz',\n",
    "                remove_unkown=True,\n",
    "                remove_phased_gt=True,\n",
    "                only_peak_variants=True)\n",
    "sinclair_sample_data = pd.read_csv('data/pedigree_data.tsv', sep='\\t')\n",
    "sin_probands = sinclair_sample_data.id[sinclair_sample_data.fam_relation == 0].tolist() \n",
    "sin_non_probands = sinclair_sample_data.id[~sinclair_sample_data.id.isin(probands)] \n",
    "ken_probands =   [i for i in gdf.get_samples() if i.startswith('P') and i != 'PKGen169']\n",
    "ken_non_probands = ['PKGen169']\n",
    "interval_df = pd.read_csv('data/intervals_with_overlaps_1_1.5mb.csv', index_col=0).set_index('INTERVAL_ID')\n",
    "# gdf.get_table().head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "028e3229",
   "metadata": {},
   "source": [
    "We will start with sinclair data <br>\n",
    "first, we need to filter the main table to match the samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "fc924288",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sincalair data contains 9 samples with DSD and 15 without \n",
      " total- 24 variants\n"
     ]
    }
   ],
   "source": [
    "print(f'Sincalair data contains {len(sin_probands)} samples with DSD and \\\n",
    "{len(sin_non_probands)} without \\n total- {len(sin_probands) +len(sin_non_probands)} variants')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "4116126f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering samples\n",
      "removing variants outside of peak interval\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "done\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>NEW_AF</th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009</th>\n",
       "      <th>AS22WG010</th>\n",
       "      <th>AS22WG011</th>\n",
       "      <th>AS22WG023</th>\n",
       "      <th>AS22WG024</th>\n",
       "      <th>AS22WG004</th>\n",
       "      <th>AS22WG005</th>\n",
       "      <th>AS22WG006</th>\n",
       "      <th>...</th>\n",
       "      <th>AS22WG013</th>\n",
       "      <th>AS22WG014</th>\n",
       "      <th>AS22WG015</th>\n",
       "      <th>AS22WG016</th>\n",
       "      <th>AS22WG017</th>\n",
       "      <th>AS22WG018</th>\n",
       "      <th>AS22WG019</th>\n",
       "      <th>AS22WG020</th>\n",
       "      <th>AS22WG021</th>\n",
       "      <th>AS22WG022</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>634112</th>\n",
       "      <th>T</th>\n",
       "      <th>C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>PGRA.SER.GRA_1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778639</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>0.127933</td>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779047</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>0.819806</td>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826893</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>0.730133</td>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827209</th>\n",
       "      <th>G</th>\n",
       "      <th>C</th>\n",
       "      <td>0.739536</td>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        NEW_AF          INTERVAL_ID  AS22WG009  AS22WG010  \\\n",
       "CHROM POS    REF ALT                                                        \n",
       "chr1  634112 T   C         NaN       PGRA.SER.GRA_1       True       True   \n",
       "      778639 A   G    0.127933  PGRA.SER.GRA.PSER_2      False      False   \n",
       "      779047 G   A    0.819806  PGRA.SER.GRA.PSER_2       True       True   \n",
       "      826893 G   A    0.730133  PGRA.SER.GRA.PSER_3       True       True   \n",
       "      827209 G   C    0.739536  PGRA.SER.GRA.PSER_3       True       True   \n",
       "\n",
       "                      AS22WG011  AS22WG023  AS22WG024  AS22WG004  AS22WG005  \\\n",
       "CHROM POS    REF ALT                                                          \n",
       "chr1  634112 T   C         True       True       True       True       True   \n",
       "      778639 A   G        False      False      False      False      False   \n",
       "      779047 G   A         True       True       True       True       True   \n",
       "      826893 G   A         True       True       True       True       True   \n",
       "      827209 G   C         True       True       True       True       True   \n",
       "\n",
       "                      AS22WG006  ...  AS22WG013  AS22WG014  AS22WG015  \\\n",
       "CHROM POS    REF ALT             ...                                    \n",
       "chr1  634112 T   C         True  ...       True       True       True   \n",
       "      778639 A   G        False  ...      False      False      False   \n",
       "      779047 G   A         True  ...       True       True       True   \n",
       "      826893 G   A         True  ...       True       True       True   \n",
       "      827209 G   C         True  ...       True       True       True   \n",
       "\n",
       "                      AS22WG016  AS22WG017  AS22WG018  AS22WG019  AS22WG020  \\\n",
       "CHROM POS    REF ALT                                                          \n",
       "chr1  634112 T   C         True       True       True       True       True   \n",
       "      778639 A   G        False      False      False      False      False   \n",
       "      779047 G   A         True       True       True       True       True   \n",
       "      826893 G   A         True       True       True       True       True   \n",
       "      827209 G   C         True       True       True       True       True   \n",
       "\n",
       "                      AS22WG021  AS22WG022  \n",
       "CHROM POS    REF ALT                        \n",
       "chr1  634112 T   C         True       True  \n",
       "      778639 A   G        False      False  \n",
       "      779047 G   A         True       True  \n",
       "      826893 G   A         True       True  \n",
       "      827209 G   C         True       True  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdf.filter_samples([i for i in gdf.get_samples() if i not in sin_probands + sin_non_probands.tolist() ])\n",
    "bool_df = gdf.bool_variant_df()\n",
    "bool_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "45020d9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_numbers(bool_df,probands, non_probands, prefix='', AF_t=1):\n",
    "    bool_df = bool_df[bool_df.NEW_AF <= AF_t]\n",
    "    all_samples = bool_df.drop(columns='NEW_AF').groupby('INTERVAL_ID').sum().sum(axis=1).rename(prefix + 'all_samples')\n",
    "    only_probands = bool_df.drop(columns=['NEW_AF'] + non_probands).groupby('INTERVAL_ID').sum().sum(axis=1).rename(prefix + 'only_probands')\n",
    "    only_non_probands = bool_df.drop(columns=['NEW_AF'] + probands).groupby('INTERVAL_ID').sum().sum(axis=1).rename(prefix +'only_non_probands')\n",
    "    return pd.DataFrame(data=[all_samples, only_probands, only_non_probands]).T\n",
    "\n",
    "# sin_af1  = get_numbers(bool_df, sin_probands, sin_non_probands.tolist(),'sin_').sort_values('sin_all_samples', ascending=False)\n",
    "# sin_aff =  get_numbers(bool_df, sin_probands, sin_non_probands.tolist(),'sin_af<0.01_', AF_t=0.01).sort_values('sin_af<0.01_all_samples', ascending=False)\n",
    "# sin_aff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "7a9df1f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "sinclair = pd.concat([interval_df,sin_af1,sin_aff], axis=1).sort_values('sin_af<0.01_all_samples', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "856045f5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ken_af&lt;0.01_all_samples</th>\n",
       "      <th>ken_af&lt;0.01_only_probands</th>\n",
       "      <th>ken_af&lt;0.01_only_non_probands</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CM100017</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM100097</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM100099</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM100102</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM100109</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             ken_af<0.01_all_samples  ken_af<0.01_only_probands  \\\n",
       "INTERVAL_ID                                                       \n",
       "CM100017                           1                          1   \n",
       "CM100097                           1                          1   \n",
       "CM100099                           1                          0   \n",
       "CM100102                           1                          1   \n",
       "CM100109                           1                          1   \n",
       "\n",
       "             ken_af<0.01_only_non_probands  \n",
       "INTERVAL_ID                                 \n",
       "CM100017                                 0  \n",
       "CM100097                                 0  \n",
       "CM100099                                 1  \n",
       "CM100102                                 0  \n",
       "CM100109                                 0  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# gdf.reset_table().filter_samples([i for i in gdf.get_samples() if i not in ken_probands + ken_non_probands ])\n",
    "# bool_df = gdf.bool_variant_df()\n",
    "# bool_df.head()\n",
    "\n",
    "ken_af1  = get_numbers(bool_df, ken_probands, ken_non_probands,'ken_')\n",
    "ken_aff =  get_numbers(bool_df, ken_probands, ken_non_probands,'ken_af<0.01_', AF_t=0.01)\n",
    "ken_aff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "1b3fd34b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>FROM</th>\n",
       "      <th>TO</th>\n",
       "      <th>GH_DSD_genes</th>\n",
       "      <th>GH_DSD</th>\n",
       "      <th>dsd_genes1mb</th>\n",
       "      <th>dsd_gene1.5mb</th>\n",
       "      <th>all_genes1.5mb</th>\n",
       "      <th>num_of_genes</th>\n",
       "      <th>ken_all_samples</th>\n",
       "      <th>ken_only_probands</th>\n",
       "      <th>ken_only_non_probands</th>\n",
       "      <th>ken_af&lt;0.01_all_samples</th>\n",
       "      <th>ken_af&lt;0.01_only_probands</th>\n",
       "      <th>ken_af&lt;0.01_only_non_probands</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PGRA.SER_6229</th>\n",
       "      <td>chr11</td>\n",
       "      <td>1015808</td>\n",
       "      <td>1017938</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGF2</td>\n",
       "      <td>WBP1LP10_OR4F2P_ENSG00000254468_CICP23_LINC010...</td>\n",
       "      <td>132</td>\n",
       "      <td>74.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mATAC_hATAC_32167</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31355973</td>\n",
       "      <td>31357525</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000290870_HCP5B_ENSG00000230521_HLA-H_HL...</td>\n",
       "      <td>210</td>\n",
       "      <td>66.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mATAC_hATAC_32222</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32583486</td>\n",
       "      <td>32584897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C6orf15_PSORS1C1_CDSN_PSORS1C2_POLR2LP1_CCHCR1...</td>\n",
       "      <td>186</td>\n",
       "      <td>77.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mATAC_hATAC_32164</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31270881</td>\n",
       "      <td>31272606</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-V_HCG4_HLA-V_HLA-P_RPL7AP7_MICG_HLA-G_HCG4...</td>\n",
       "      <td>215</td>\n",
       "      <td>192.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGRA.SER.PSER.GRA_28538</th>\n",
       "      <td>chr3</td>\n",
       "      <td>75667918</td>\n",
       "      <td>75670704</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CNTN3_NIPA2P2_RN7SL294P_MIR4444-2_ENSG00000287...</td>\n",
       "      <td>41</td>\n",
       "      <td>69.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SER_42024</th>\n",
       "      <td>chrY</td>\n",
       "      <td>19743944</td>\n",
       "      <td>19745337</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAF9P2_CLUHP2_FAM224A_ENSG00000278391_RNA5SP52...</td>\n",
       "      <td>33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SER.PSER_42025</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20574998</td>\n",
       "      <td>20576299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTTY14_ENSG00000274282_GAPDHP19_BCORP1_BCORP1_...</td>\n",
       "      <td>38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM189619</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20753080</td>\n",
       "      <td>20753096</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000274282_GAPDHP19_BCORP1_BCORP1_TXLNGY_...</td>\n",
       "      <td>38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM189620</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57183942</td>\n",
       "      <td>57184105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CM189621</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57184209</td>\n",
       "      <td>57184541</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193680 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         CHROM      FROM        TO GH_DSD_genes  GH_DSD  \\\n",
       "INTERVAL_ID                                                               \n",
       "PGRA.SER_6229            chr11   1015808   1017938          NaN   False   \n",
       "mATAC_hATAC_32167         chr6  31355973  31357525          NaN   False   \n",
       "mATAC_hATAC_32222         chr6  32583486  32584897          NaN   False   \n",
       "mATAC_hATAC_32164         chr6  31270881  31272606          NaN   False   \n",
       "PGRA.SER.PSER.GRA_28538   chr3  75667918  75670704          NaN   False   \n",
       "...                        ...       ...       ...          ...     ...   \n",
       "SER_42024                 chrY  19743944  19745337          NaN   False   \n",
       "SER.PSER_42025            chrY  20574998  20576299          NaN   False   \n",
       "CM189619                  chrY  20753080  20753096          NaN   False   \n",
       "CM189620                  chrY  57183942  57184105          NaN   False   \n",
       "CM189621                  chrY  57184209  57184541          NaN   False   \n",
       "\n",
       "                        dsd_genes1mb dsd_gene1.5mb  \\\n",
       "INTERVAL_ID                                          \n",
       "PGRA.SER_6229                    NaN          IGF2   \n",
       "mATAC_hATAC_32167                NaN           NaN   \n",
       "mATAC_hATAC_32222                NaN           NaN   \n",
       "mATAC_hATAC_32164                NaN           NaN   \n",
       "PGRA.SER.PSER.GRA_28538          NaN           NaN   \n",
       "...                              ...           ...   \n",
       "SER_42024                        NaN           NaN   \n",
       "SER.PSER_42025                   NaN           NaN   \n",
       "CM189619                         NaN           NaN   \n",
       "CM189620                         NaN           NaN   \n",
       "CM189621                         NaN           NaN   \n",
       "\n",
       "                                                            all_genes1.5mb  \\\n",
       "INTERVAL_ID                                                                  \n",
       "PGRA.SER_6229            WBP1LP10_OR4F2P_ENSG00000254468_CICP23_LINC010...   \n",
       "mATAC_hATAC_32167        ENSG00000290870_HCP5B_ENSG00000230521_HLA-H_HL...   \n",
       "mATAC_hATAC_32222        C6orf15_PSORS1C1_CDSN_PSORS1C2_POLR2LP1_CCHCR1...   \n",
       "mATAC_hATAC_32164        HLA-V_HCG4_HLA-V_HLA-P_RPL7AP7_MICG_HLA-G_HCG4...   \n",
       "PGRA.SER.PSER.GRA_28538  CNTN3_NIPA2P2_RN7SL294P_MIR4444-2_ENSG00000287...   \n",
       "...                                                                    ...   \n",
       "SER_42024                TAF9P2_CLUHP2_FAM224A_ENSG00000278391_RNA5SP52...   \n",
       "SER.PSER_42025           TTTY14_ENSG00000274282_GAPDHP19_BCORP1_BCORP1_...   \n",
       "CM189619                 ENSG00000274282_GAPDHP19_BCORP1_BCORP1_TXLNGY_...   \n",
       "CM189620                 CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...   \n",
       "CM189621                 CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...   \n",
       "\n",
       "                         num_of_genes  ken_all_samples  ken_only_probands  \\\n",
       "INTERVAL_ID                                                                 \n",
       "PGRA.SER_6229                     132             74.0               69.0   \n",
       "mATAC_hATAC_32167                 210             66.0               60.0   \n",
       "mATAC_hATAC_32222                 186             77.0               75.0   \n",
       "mATAC_hATAC_32164                 215            192.0              171.0   \n",
       "PGRA.SER.PSER.GRA_28538            41             69.0               66.0   \n",
       "...                               ...              ...                ...   \n",
       "SER_42024                          33              NaN                NaN   \n",
       "SER.PSER_42025                     38              NaN                NaN   \n",
       "CM189619                           38              NaN                NaN   \n",
       "CM189620                           11              NaN                NaN   \n",
       "CM189621                           12              NaN                NaN   \n",
       "\n",
       "                         ken_only_non_probands  ken_af<0.01_all_samples  \\\n",
       "INTERVAL_ID                                                               \n",
       "PGRA.SER_6229                              5.0                     54.0   \n",
       "mATAC_hATAC_32167                          6.0                     40.0   \n",
       "mATAC_hATAC_32222                          2.0                     36.0   \n",
       "mATAC_hATAC_32164                         21.0                     35.0   \n",
       "PGRA.SER.PSER.GRA_28538                    3.0                     32.0   \n",
       "...                                        ...                      ...   \n",
       "SER_42024                                  NaN                      NaN   \n",
       "SER.PSER_42025                             NaN                      NaN   \n",
       "CM189619                                   NaN                      NaN   \n",
       "CM189620                                   NaN                      NaN   \n",
       "CM189621                                   NaN                      NaN   \n",
       "\n",
       "                         ken_af<0.01_only_probands  \\\n",
       "INTERVAL_ID                                          \n",
       "PGRA.SER_6229                                 53.0   \n",
       "mATAC_hATAC_32167                             37.0   \n",
       "mATAC_hATAC_32222                             34.0   \n",
       "mATAC_hATAC_32164                             31.0   \n",
       "PGRA.SER.PSER.GRA_28538                       31.0   \n",
       "...                                            ...   \n",
       "SER_42024                                      NaN   \n",
       "SER.PSER_42025                                 NaN   \n",
       "CM189619                                       NaN   \n",
       "CM189620                                       NaN   \n",
       "CM189621                                       NaN   \n",
       "\n",
       "                         ken_af<0.01_only_non_probands  \n",
       "INTERVAL_ID                                             \n",
       "PGRA.SER_6229                                      1.0  \n",
       "mATAC_hATAC_32167                                  3.0  \n",
       "mATAC_hATAC_32222                                  2.0  \n",
       "mATAC_hATAC_32164                                  4.0  \n",
       "PGRA.SER.PSER.GRA_28538                            1.0  \n",
       "...                                                ...  \n",
       "SER_42024                                          NaN  \n",
       "SER.PSER_42025                                     NaN  \n",
       "CM189619                                           NaN  \n",
       "CM189620                                           NaN  \n",
       "CM189621                                           NaN  \n",
       "\n",
       "[193680 rows x 15 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ken"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "7b9832e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "ken = pd.concat([interval_df,ken_af1,ken_aff], axis=1).sort_values('ken_af<0.01_all_samples', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "a944db87",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# create a excel writer object\n",
    "with pd.ExcelWriter(\"outputs/interval_variants.xlsx\") as writer:\n",
    "   \n",
    "    # use to_excel function and specify the sheet_name and index\n",
    "    # to store the dataframe in specified sheet\n",
    "    sinclair.to_excel(writer, sheet_name=\"sinclair\")\n",
    "    ken.to_excel(writer, sheet_name=\"ken\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "e0e58c75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>FROM</th>\n",
       "      <th>TO</th>\n",
       "      <th>GH_DSD_genes</th>\n",
       "      <th>GH_DSD</th>\n",
       "      <th>dsd_genes1mb</th>\n",
       "      <th>dsd_gene1.5mb</th>\n",
       "      <th>all_genes1.5mb</th>\n",
       "      <th>num_of_genes</th>\n",
       "      <th>sin_all_samples</th>\n",
       "      <th>sin_only_probands</th>\n",
       "      <th>sin_only_non_probands</th>\n",
       "      <th>sin_af&lt;0.01_all_samples</th>\n",
       "      <th>sin_af&lt;0.01_only_probands</th>\n",
       "      <th>sin_af&lt;0.01_only_non_probands</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PGRA.SER_6229</td>\n",
       "      <td>chr11</td>\n",
       "      <td>1015808</td>\n",
       "      <td>1017938</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGF2</td>\n",
       "      <td>WBP1LP10_OR4F2P_ENSG00000254468_CICP23_LINC010...</td>\n",
       "      <td>132</td>\n",
       "      <td>70.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PGRA.SER.PSER.GRA_28538</td>\n",
       "      <td>chr3</td>\n",
       "      <td>75667918</td>\n",
       "      <td>75670704</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CNTN3_NIPA2P2_RN7SL294P_MIR4444-2_ENSG00000287...</td>\n",
       "      <td>41</td>\n",
       "      <td>122.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PGRA.SER.GRA.PSER_39577</td>\n",
       "      <td>chr9</td>\n",
       "      <td>62837274</td>\n",
       "      <td>62838738</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CNTNAP3C_RN7SL462P_ENSG00000277109_FAM242D_Y_R...</td>\n",
       "      <td>65</td>\n",
       "      <td>79.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mATAC_hATAC_22657</td>\n",
       "      <td>chr2</td>\n",
       "      <td>178193444</td>\n",
       "      <td>178195490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LINC01117_ENSG00000230552_FUCA1P1_ENSG00000223...</td>\n",
       "      <td>52</td>\n",
       "      <td>68.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mATAC_hATAC_9960</td>\n",
       "      <td>chr12</td>\n",
       "      <td>108855616</td>\n",
       "      <td>108858272</td>\n",
       "      <td>['SART3 : 23.5516']</td>\n",
       "      <td>True</td>\n",
       "      <td>SART3</td>\n",
       "      <td>SART3</td>\n",
       "      <td>ABTB3_ENSG00000200897_RNA5SP371_ENSG0000025757...</td>\n",
       "      <td>53</td>\n",
       "      <td>33.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193675</th>\n",
       "      <td>SER_42024</td>\n",
       "      <td>chrY</td>\n",
       "      <td>19743944</td>\n",
       "      <td>19745337</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAF9P2_CLUHP2_FAM224A_ENSG00000278391_RNA5SP52...</td>\n",
       "      <td>33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193676</th>\n",
       "      <td>SER.PSER_42025</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20574998</td>\n",
       "      <td>20576299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TTTY14_ENSG00000274282_GAPDHP19_BCORP1_BCORP1_...</td>\n",
       "      <td>38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193677</th>\n",
       "      <td>CM189619</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20753080</td>\n",
       "      <td>20753096</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000274282_GAPDHP19_BCORP1_BCORP1_TXLNGY_...</td>\n",
       "      <td>38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193678</th>\n",
       "      <td>CM189620</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57183942</td>\n",
       "      <td>57184105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193679</th>\n",
       "      <td>CM189621</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57184209</td>\n",
       "      <td>57184541</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193680 rows Ã— 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    INTERVAL_ID  CHROM       FROM         TO  \\\n",
       "0                 PGRA.SER_6229  chr11    1015808    1017938   \n",
       "1       PGRA.SER.PSER.GRA_28538   chr3   75667918   75670704   \n",
       "2       PGRA.SER.GRA.PSER_39577   chr9   62837274   62838738   \n",
       "3             mATAC_hATAC_22657   chr2  178193444  178195490   \n",
       "4              mATAC_hATAC_9960  chr12  108855616  108858272   \n",
       "...                         ...    ...        ...        ...   \n",
       "193675                SER_42024   chrY   19743944   19745337   \n",
       "193676           SER.PSER_42025   chrY   20574998   20576299   \n",
       "193677                 CM189619   chrY   20753080   20753096   \n",
       "193678                 CM189620   chrY   57183942   57184105   \n",
       "193679                 CM189621   chrY   57184209   57184541   \n",
       "\n",
       "               GH_DSD_genes  GH_DSD dsd_genes1mb dsd_gene1.5mb  \\\n",
       "0                       NaN   False          NaN          IGF2   \n",
       "1                       NaN   False          NaN           NaN   \n",
       "2                       NaN   False          NaN           NaN   \n",
       "3                       NaN   False          NaN           NaN   \n",
       "4       ['SART3 : 23.5516']    True        SART3         SART3   \n",
       "...                     ...     ...          ...           ...   \n",
       "193675                  NaN   False          NaN           NaN   \n",
       "193676                  NaN   False          NaN           NaN   \n",
       "193677                  NaN   False          NaN           NaN   \n",
       "193678                  NaN   False          NaN           NaN   \n",
       "193679                  NaN   False          NaN           NaN   \n",
       "\n",
       "                                           all_genes1.5mb  num_of_genes  \\\n",
       "0       WBP1LP10_OR4F2P_ENSG00000254468_CICP23_LINC010...           132   \n",
       "1       CNTN3_NIPA2P2_RN7SL294P_MIR4444-2_ENSG00000287...            41   \n",
       "2       CNTNAP3C_RN7SL462P_ENSG00000277109_FAM242D_Y_R...            65   \n",
       "3       LINC01117_ENSG00000230552_FUCA1P1_ENSG00000223...            52   \n",
       "4       ABTB3_ENSG00000200897_RNA5SP371_ENSG0000025757...            53   \n",
       "...                                                   ...           ...   \n",
       "193675  TAF9P2_CLUHP2_FAM224A_ENSG00000278391_RNA5SP52...            33   \n",
       "193676  TTTY14_ENSG00000274282_GAPDHP19_BCORP1_BCORP1_...            38   \n",
       "193677  ENSG00000274282_GAPDHP19_BCORP1_BCORP1_TXLNGY_...            38   \n",
       "193678  CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...            11   \n",
       "193679  CTBP2P1_AMD1P2_DPH3P2_VAMP7_ELOCP24_TRPC6P1_IL...            12   \n",
       "\n",
       "        sin_all_samples  sin_only_probands  sin_only_non_probands  \\\n",
       "0                  70.0               32.0                   38.0   \n",
       "1                 122.0               53.0                   69.0   \n",
       "2                  79.0               32.0                   47.0   \n",
       "3                  68.0               26.0                   42.0   \n",
       "4                  33.0               13.0                   20.0   \n",
       "...                 ...                ...                    ...   \n",
       "193675              NaN                NaN                    NaN   \n",
       "193676              NaN                NaN                    NaN   \n",
       "193677              NaN                NaN                    NaN   \n",
       "193678              NaN                NaN                    NaN   \n",
       "193679              NaN                NaN                    NaN   \n",
       "\n",
       "        sin_af<0.01_all_samples  sin_af<0.01_only_probands  \\\n",
       "0                          38.0                       15.0   \n",
       "1                          33.0                       10.0   \n",
       "2                          31.0                       14.0   \n",
       "3                          25.0                       10.0   \n",
       "4                           3.0                        1.0   \n",
       "...                         ...                        ...   \n",
       "193675                      NaN                        NaN   \n",
       "193676                      NaN                        NaN   \n",
       "193677                      NaN                        NaN   \n",
       "193678                      NaN                        NaN   \n",
       "193679                      NaN                        NaN   \n",
       "\n",
       "        sin_af<0.01_only_non_probands  \n",
       "0                                23.0  \n",
       "1                                23.0  \n",
       "2                                17.0  \n",
       "3                                15.0  \n",
       "4                                 2.0  \n",
       "...                               ...  \n",
       "193675                            NaN  \n",
       "193676                            NaN  \n",
       "193677                            NaN  \n",
       "193678                            NaN  \n",
       "193679                            NaN  \n",
       "\n",
       "[193680 rows x 16 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "dev-venv",
   "language": "python",
   "name": "dev-venv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
