{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8af24c60",
   "metadata": {},
   "source": [
    "# GnomAD_df class use example\n",
    "After each cell There is an explanation of what it is doing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "334ff999",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (444435471.py, line 60)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"/tmp/ipykernel_32740/444435471.py\"\u001b[0;36m, line \u001b[0;32m60\u001b[0m\n\u001b[0;31m    def add_data(self,str path,original=True,header=True, sep='\\t', list: columns=None):\u001b[0m\n\u001b[0m                             ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "class GnomAD_df:\n",
    "        \n",
    "        \n",
    "    \n",
    "    def __init__(self,path, peak_file=None, remove_unkown=True, remove_phased_gt=True, only_peak_variants=True):\n",
    "        \"\"\"\n",
    "        This class warps the result of gnomAD and interval  mATAC/hATAC analysis.\n",
    "        path - path to the parquet dataframe file \n",
    "        peak_file - bed file which contains the following columns - CHROM,FROM,TO,INTERVAL_ID, ~BUT w/o header~\n",
    "        remove_unknown - removed unkwon genotypes (e.g \"./.\")' \n",
    "        remove_phased_gt - \"replaced phased genotype (e.g \"0|1\" -> \"0/1\")\"\n",
    "        only_peak_variants - removing variants outside of peak interval\n",
    "        \n",
    "        \"\"\"\n",
    "        self.__original_df = self.__open_table(path)\n",
    "        self.f_phasing = False\n",
    "        self.__filter_samples = []\n",
    "        self.__filters_description = []\n",
    "        self.__filter_functions = []\n",
    "        self.__cur_df = self.__original_df.copy()\n",
    "        self.__applied_filter_index = 0\n",
    "        if only_peak_variants:\n",
    "            self.remove_non_peak_variants()\n",
    "        if remove_unkown:\n",
    "            self.remove_unkown()\n",
    "        if remove_phased_gt:\n",
    "            self.remove_phasing()\n",
    "        self.__samples = [i.split(':')[0] for i in self.__original_df.columns if i.endswith('GT')]\n",
    "        if peak_file != None:\n",
    "            self.__peak_df = self.__get_peak_df(peak_file)\n",
    "    \n",
    "    \n",
    "    def __get_peak_df(self, path):\n",
    "        peak_df = pd.read_csv(path, sep='\\t', header=None)\n",
    "        peak_df.columns = ['CHROM', 'FROM','TO', 'INTERVAL_ID']\n",
    "        return peak_df\n",
    "    \n",
    "    def load_peak_file(peak_file_path):\n",
    "        \"\"\"\n",
    "        loads a peak file into the object\n",
    "        \"\"\"\n",
    "        self.__peak_df = self.__get_peak_df(peak_file)\n",
    "    \n",
    "    def get_peak_df(self):\n",
    "        \"\"\"\n",
    "        returns the intervals dataframe \n",
    "        \"\"\"\n",
    "        return self.__peak_df\n",
    "    \n",
    "    def __open_table(self, path):\n",
    "        df = pd.read_parquet(path).fillna(value=np.nan).replace('0/0',np.nan).replace('0|0',np.nan).drop(columns='AF')\n",
    "        interval = df.INTERVAL_ID.copy()\n",
    "        df = df.drop(columns='INTERVAL_ID')\n",
    "        df.insert(0,'INTERVAL_ID',interval)\n",
    "        return df\n",
    "    \n",
    "    \n",
    "    def add_data(self,str: path,original=True,header=True, sep='\\t', list: columns=None):\n",
    "        \"\"\"\n",
    "        add the dataframe from the file / columns which were provided in a list (columns) (If none - all columns will be added)\n",
    "        the file need to have CHROM,POS, REF,ALT\n",
    "        if original = True, adds the data to the original otherwise to the current df\n",
    "        \"\"\"\n",
    "        to_add = pd.read_csv(path, sep=sep, header=True).set_index(['CHROM','POS', 'REF','ALT'])\n",
    "        if not original:\n",
    "        add_to = self.__original_df.copy() if original  else self.self.__cur_df\n",
    "        # todo - understand how to integraye the new data- should i uuse reset table.\n",
    "\n",
    "        \n",
    "    def __remove_non_peak_variants(self):\n",
    "\n",
    "        self.__cur_df = self.__cur_df.replace('.', np.nan)\n",
    "        self.__cur_df = self.__cur_df[~self.__cur_df.INTERVAL_ID.isna()]\n",
    "\n",
    "    def remove_non_peak_variants(self):\n",
    "        \"\"\"\n",
    "        remove variants if they are not in a peak\n",
    "        \"\"\"\n",
    "        decs=\"removing variants outside of peak interval\"\n",
    "        \n",
    "        if self.__check_if_filter_exists(decs):\n",
    "            self.__filters_description.append(\"removing variants outside of peak interval\")\n",
    "            self.__filter_functions.append(self.__remove_non_peak_variants)\n",
    "        return self\n",
    "\n",
    "    def __remove_unknow(self):\n",
    "        self.__cur_df = self.__cur_df.replace('./.', np.nan).replace('.|.', np.nan)\n",
    "\n",
    "    def remove_unkown(self):\n",
    "        \"\"\"\n",
    "        removed unkwon genotypes (e.g \"./.\")\n",
    "        \"\"\"\n",
    "        decs='removed unkwon genotypes (e.g \\\"./.\\\")'\n",
    "        \n",
    "        if self.__check_if_filter_exists(decs):\n",
    "            self.__filters_description.append('removed unkwon genotypes (e.g \\\"./.\\\")')\n",
    "            self.__filter_functions.append(self.__remove_unknow)\n",
    "        return self\n",
    "\n",
    "    def remove_phasing(self):\n",
    "        \"\"\"\n",
    "        \"replaced phased genotype (e.g \"0|1\" -> \"0/1\")\"\n",
    "        \"\"\"\n",
    "        desc = \"replaced phased genotype (e.g \\\"0|1\\\" -> \\\"0/1\\\")\"\n",
    "        if self.__check_if_filter_exists(desc):\n",
    "            self.__filters_description.append(desc)\n",
    "            self.__filter_functions.append(self.__remove_phasing)\n",
    "        return self\n",
    "\n",
    "    def __remove_phasing(self):\n",
    "\n",
    "        \"\"\"\n",
    "        replace all \"X|X\" into \"X/X\"\n",
    "        \"\"\"\n",
    "        if self.f_phasing: return\n",
    "        for col in [i for i in self.__cur_df.columns if i.endswith('GT')]:\n",
    "            for val in self.__cur_df[col].unique():\n",
    "                if val == np.nan: continue\n",
    "                if '|' in str(val):\n",
    "                    self.__cur_df = self.__cur_df.replace(val, \"%s/%s\" % tuple(val.split('|')))\n",
    "        self.f_phasing = True\n",
    "    \n",
    "    \n",
    "    def __check_if_filter_exists(self,description):\n",
    "        \"\"\"\n",
    "        Indicate if a filter already applied on the DF by checking if the description exist in the list.\n",
    "        Return True if filter was not applied\n",
    "        \"\"\"\n",
    "        if description in self.__filters_description:\n",
    "            print(f\"{description.split('(')[0]} already applied\")\n",
    "            return False\n",
    "        return True\n",
    "        \n",
    "        \n",
    "    \n",
    "    def filter_DP(self, dp_t:float):\n",
    "        \"\"\"\n",
    "        Fiter the variants (replace the GT with NaN) according to a given threshold (dp_t)\n",
    "        \"\"\"\n",
    "        desc = f\"Removing variants with reading depth {dp_t} or below\"\n",
    "        if self.__check_if_filter_exists(desc):\n",
    "            self.__filters_description.append(desc)\n",
    "            self.__filter_functions.append(lambda: self.__filter_dp(dp_t))\n",
    "        return self\n",
    "\n",
    "    def __filter_dp(self, dp_t:float):\n",
    "        for sample in self.__samples:\n",
    "            self.__cur_df.loc[self.__cur_df[sample + ':DP'] <= dp_t, sample + ':GT'] = np.nan\n",
    "\n",
    "    def ___filter_AF_remove_unknown(self, af_t:float):\n",
    "        self.__cur_df = self.__cur_df[(self.__cur_df.AF <= af_t) | (self.__cur_df.AF.isna())]\n",
    "\n",
    "    def ___filter_AF(self,af_t:float):\n",
    "        self.__cur_df = self.__cur_df[(self.__cur_df.AF <= af_t)]\n",
    "\n",
    "    def filter_AF(self, af_t: float,remove_unkwon=False):\n",
    "        \"\"\"\n",
    "        Filters the variants of gnomAD's allele frequency (AF) below a given threshold.\n",
    "        If remove_unkown is True, remove variants with no record on gnomAD\n",
    "        \"\"\"\n",
    "        description = f\"Removing variants with allele frequency above {af_t}\"\n",
    "        description += \" and variants with no record on gnomAD\" if remove_unkwon else \"\"\n",
    "        if self.__check_if_filter_exists(description):\n",
    "            self.__filters_description.append(description)\n",
    "            if not remove_unkwon:\n",
    "                self.__filter_functions.append(lambda :self.___filter_AF_remove_unknown(af_t))\n",
    "            else:\n",
    "                self.__filter_functions.append(lambda :self.___filter_AF(af_t))\n",
    "        return self\n",
    "\n",
    "    def samples_filter(self,sample_list, appened=True):\n",
    "        \"\"\"\n",
    "        Filter the variants dataframe to a given set of samples.\n",
    "        If appened is False, remove all previous given samples of filteration\n",
    "\n",
    "        \"\"\"\n",
    "        if appened:\n",
    "            self.filter_samples += sample_list\n",
    "        else:\n",
    "            self.filter_samples = sample_list\n",
    "        return self\n",
    "\n",
    "    def __sample_filter(self):\n",
    "        to_remove_samples = [i for i in self.__samples if i in self.__filter_samples]\n",
    "        drop_cols = []\n",
    "        for i in to_remove_samples:\n",
    "            self.__cur_df = self.__cur_df.drop(columns=[f'{i}:GT', f'{i}:DP'])\n",
    "\n",
    "    def print_filters(self):\n",
    "        \"\"\"\n",
    "        prints the filters applied on the variant df\n",
    "        \"\"\"\n",
    "        for f in range(len(self.__filters_description)):\n",
    "            print(f\"{f}. {self.__filters_description[f]}\")\n",
    "\n",
    "    def remove_filter(self, i):\n",
    "        \"\"\"\n",
    "        remove the i'th filter from the table \n",
    "        To know the filters index, use \"print_filter\"\n",
    "        \"\"\"\n",
    "        if i >= len(self.__filters_description):\n",
    "            print(\"Filter does not exist\")\n",
    "            return self\n",
    "        del self.__filters_description[i]\n",
    "        del self.__filter_functions[i]\n",
    "        if i <= self.__applied_filter_index:\n",
    "            self.__cur_df = self.__original_df.copy()\n",
    "            self.f_phasing = False\n",
    "            self.__applied_filter_index = 0\n",
    "        return self\n",
    "\n",
    "    def reset_table(self, remove_phased_gt=True, remove_unkown=True, only_peak_variants=True):\n",
    "        \"\"\"\n",
    "        returns the dataframe to it's initial state.\n",
    "        removes all filters applied on the dataframe\n",
    "        \"\"\"\n",
    "        self.__filter_samples = []\n",
    "        self.__filters_description = []\n",
    "        self.__filter_functions = []\n",
    "        self.__applied_filter_index = 0\n",
    "        self.__samples = [i.split(':')[0] for i in self.__original_df.columns if i.endswith('GT')]\n",
    "        self.__cur_df = self.__original_df.copy()\n",
    "        self.f_phasing = False\n",
    "        if only_peak_variants:\n",
    "            self.remove_non_peak_variants()\n",
    "        if remove_phased_gt:\n",
    "            self.remove_phasing()\n",
    "        if remove_unkown:\n",
    "            self.remove_unkown()\n",
    "        return self\n",
    "\n",
    "    def get_filters_decription(self):\n",
    "        \"\"\"\n",
    "        returns list of filter description\n",
    "        \"\"\"\n",
    "        return self.__filters_description\n",
    "        \n",
    "    def __apply_filters(self, verbos=True):\n",
    "        max_index = len(self.__filters_description)\n",
    "        # filter samples only if there are samples in the list\n",
    "        if len(self.__filter_samples):\n",
    "            if verbos:\n",
    "                print(\"Filtering samples\")\n",
    "            self.__sample_filter()\n",
    "        if verbos:\n",
    "            for i in range(self.__applied_filter_index):\n",
    "                print(self.__filters_description[i])\n",
    "        for i in range(self.__applied_filter_index, max_index):\n",
    "            if verbos:\n",
    "                print(self.__filters_description[i])\n",
    "            self.__filter_functions[i]()\n",
    "        print(\"done\")\n",
    "        self.__applied_filter_index = max_index\n",
    "\n",
    "\n",
    "    def get_table(self, verbos=True):\n",
    "        \"\"\"\n",
    "        returns the variants df filtered by all given filters.\n",
    "        Verbos = true -> prints the log of filteration\n",
    "        \"\"\"\n",
    "        if verbos:\n",
    "            print(\"applying filters\")\n",
    "        self.__apply_filters(verbos)\n",
    "        if verbos:\n",
    "            print('getting table')\n",
    "        return self.__cur_df.copy()\n",
    "\n",
    "    def __remove_reference(self):\n",
    "        \"\"\"\n",
    "        removing 0/0 and 0/. or ./0 from the table (not saving the dataframe)\n",
    "        \"\"\"\n",
    "        letters = ['0', '.']\n",
    "        df = self.__cur_df.copy()\n",
    "        for i in letters:\n",
    "            for j in letters:\n",
    "                df = df.replace(f\"{i}|{j}\", np.nan)\n",
    "        return df\n",
    "\n",
    "    def bool_variant_df(self, verbos=True):\n",
    "        \"\"\"\n",
    "        returns a boolean table of samples VS variants.\n",
    "        True if the variant exist in the sample.\n",
    "        NOTICE- THIS APPLY REMOVE PHASING FILTER ON DF\n",
    "        \"\"\"\n",
    "        self.__remove_phasing()\n",
    "        self.__apply_filters(verbos)\n",
    "        df = self.__remove_reference()\n",
    "        gt = df[[i for i in df.columns if i.endswith('GT')]].notna()\n",
    "        gt.columns = [i.replace(':GT','') for i in gt.columns ]\n",
    "        return pd.concat([df[['AF','INTERVAL_ID']],gt], axis=1)\n",
    "    \n",
    "    \n",
    "    \n",
    "    def __add_interval_info(self,df,drop_all_zero=False):\n",
    "        \"\"\"\n",
    "        assuming df index is the interval_id\n",
    "        \"\"\"\n",
    "        peak_df_isin = self.__peak_df[self.__peak_df.INTERVAL_ID.isin(df.index)].set_index('INTERVAL_ID')\n",
    "        result = pd.concat([peak_df_isin,df], axis=1)\n",
    "        if not drop_all_zero:\n",
    "            peak_df_isout = self.__peak_df[~self.__peak_df.INTERVAL_ID.isin(df.index)].set_index('INTERVAL_ID')\n",
    "            result = pd.concat([result, peak_df_isout])\n",
    "        return result\n",
    "    \n",
    "    \n",
    "    \n",
    "    def count_interval(self, boolean=False, drop_all_zero=False, per_sample=False):\n",
    "        \"\"\"\n",
    "        returns a dataframe with the amount of variant each  in each peak.\n",
    "        if boolean = True, reaturns a df of interval vs samples with Trued if sample ahve varint in that interval\n",
    "        if drop_all_zero = True, drop all interval with zero variants \n",
    "        if per_sample = True returns a dataframe with the amount of variant per sample in each peak.\n",
    "        \"\"\"\n",
    "        df =  self.bool_variant_df().drop(columns=['AF'])\n",
    "        sum_df = df.groupby('INTERVAL_ID').sum()\n",
    "        sum_df = sum_df if not boolean else sum_df != 0\n",
    "        if not per_sample:\n",
    "            sum_df = sum_df.sum(axis=1)\n",
    "        return self.__add_interval_info(sum_df,  drop_all_zero)      \n",
    "    \n",
    "    def count_peaks_variants(self,numeric=False):\n",
    "        \"\"\"\n",
    "        returns data frame with the amount of people who have variants in each interval\n",
    "        if numeric is True - the total number of variants there are in each peak of all samples\n",
    "        \"\"\"\n",
    "        bool_df = self.bool_variant_df().drop(columns=['AF'])\n",
    "        melt_df = bool_df.melt(id_vars=['INTERVAL_ID'])\n",
    "        melt_df = melt_df[melt_df.value]\n",
    "        if numeric:\n",
    "            melt_df = melt_df.groupby('INTERVAL_ID')\n",
    "        else:\n",
    "            melt_df = melt_df.drop_duplicates().groupby('INTERVAL_ID')\n",
    "        result = melt_df.value.sum().sort_values()\n",
    "        if numeric:\n",
    "            result = result.rename(columns={0:'n_variants'})\n",
    "        return \n",
    "    \n",
    "    def sum_peaks(self):\n",
    "        \"\"\"\n",
    "        sum the amount of variants in each peak \n",
    "        \"\"\"\n",
    "        bool_df = self.bool_variant_df()\n",
    "        return bool_df.groupby('INTERVAL_ID').sum()\n",
    "    \n",
    "    def get_samples(self):\n",
    "        return [i for i in self.__samples if i not in self.__filter_samples]\n",
    "    \n",
    "    def add_ken_to_main(self, to_add_path, sample_name=False, header=True, ken=False):\n",
    "        \"\"\"\n",
    "        This functions add kens tsv data to the main df\n",
    "        ken - the data is kens source\n",
    "        \"\"\"\n",
    "    \n",
    "        main_df = self.__original_df.copy()\n",
    "        if not sample_name:\n",
    "            sample_name = to_add_path.split('/')[-1].split('.')[0]\n",
    "        if sample_name in self.__samples:\n",
    "            print(f\"{sample_name} already exist\")\n",
    "            return self\n",
    "        to_add = pd.read_csv(to_add_path, sep='\\t', header=None if not header else 0)\n",
    "        col_names = [f'{sample_name}:GT',f'{sample_name}:DP']\n",
    "        columns = ['CHROM','POS', 'REF','ALT','INTERVAL_ID','AF'] + col_names\n",
    "        to_add.columns = columns\n",
    "        if ken:\n",
    "            to_add.ALT = to_add.ALT.str.replace(',<NON_REF>','')\n",
    "        to_add = to_add.replace('.',np.nan).set_index(['CHROM', 'POS', 'ALT', 'REF']).fillna(value=np.nan)\n",
    "        to_add[f'{sample_name}:DP'] = to_add[f'{sample_name}:DP'].astype(float)\n",
    "        to_add.AF = to_add.AF.astype(float)\n",
    "        exist = to_add[to_add.index.isin(main_df.index)]\n",
    "        non_exist = to_add[~to_add.index.isin(main_df.index)]\n",
    "#         return main_df,exist[col_names]\n",
    "        main_df = pd.concat([main_df,exist[col_names]],axis = 1)\n",
    "\n",
    "        main_df = pd.concat([main_df, non_exist.drop(columns=col_names)])\n",
    "        main_df.loc[non_exist.index,col_names] = non_exist[col_names]\n",
    "        self.__original_df = main_df\n",
    "        self.reset_table()\n",
    "        return self\n",
    "    \n",
    "    def count_sample_variants(self,verbos=True):\n",
    "        \"\"\"\n",
    "        Counts the amount of variant each sample have.\n",
    "        \"\"\"\n",
    "        bool_df = self.bool_variant_df(verbos=verbos)\n",
    "        return bool_df.drop(columns=['AF','INTERVAL_ID']).sum()\n",
    "    \n",
    "    def save_df(self,save_format, path, original=False):\n",
    "        \"\"\"\n",
    "        save the df in the format mentioned: csv/prq(parquet)\n",
    "        \"\"\"\n",
    "        df = self.__original_df if original else self.get_table() \n",
    "        if save_format == 'csv':\n",
    "            df.to_csv(f\"{path}.{save_format}\")\n",
    "        elif  save_format == 'prq':\n",
    "            df.to_parquet(f\"{path}.{save_format}\")\n",
    "    \n",
    "    def concat(gnomAD_df_list: list, reset_df=False, verbos=True):\n",
    "        \"\"\"\n",
    "        Generate a new gnomAD_df from a list of gnomAD_df.\n",
    "        The function takes the filtered DF from each object and creates a new one.\n",
    "        all previous filters will not be transfered, hence- the basiv DF of the new df\n",
    "        won't have the original's dfs data\n",
    "        \"\"\"\n",
    "        df_list = []\n",
    "        template = 'for samples %s the following filteres were applied:'\n",
    "        for df in gnomAD_df_list:\n",
    "            if verbos:\n",
    "                samples = \",\".join(df.get_samples())\n",
    "                print(template % samples)\n",
    "                for i in df.get_filters_decription():\n",
    "                    print(i)\n",
    "            df_list.append(df.get_table(verbos = False))\n",
    "        new_df = pd.concat(df_list, axis=1)\n",
    "        return GnomAD_df(new_df)\n",
    "    \n",
    "    \n",
    "\n",
    "                      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "13a09e1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#The first two imports are only for loading the class from it's location.\n",
    "#they are not necessery\n",
    "import sys\n",
    "import os\n",
    "sys.path.append(os.path.abspath(\"code\"))\n",
    "# these are  needed-\n",
    "import pandas as pd\n",
    "from GnomAD_df_class import GnomAD_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "52fd4f01",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "applying filters\n",
      "removing variants outside of peak interval\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "done\n",
      "getting table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009:GT</th>\n",
       "      <th>AS22WG009:DP</th>\n",
       "      <th>AS22WG010:GT</th>\n",
       "      <th>AS22WG010:DP</th>\n",
       "      <th>AS22WG011:GT</th>\n",
       "      <th>AS22WG011:DP</th>\n",
       "      <th>AS22WG023:GT</th>\n",
       "      <th>AS22WG023:DP</th>\n",
       "      <th>AS22WG024:GT</th>\n",
       "      <th>...</th>\n",
       "      <th>PKGen166:GT</th>\n",
       "      <th>PKGen166:DP</th>\n",
       "      <th>PKGen177:GT</th>\n",
       "      <th>PKGen177:DP</th>\n",
       "      <th>PKGen186:GT</th>\n",
       "      <th>PKGen186:DP</th>\n",
       "      <th>PKGen172:GT</th>\n",
       "      <th>PKGen172:DP</th>\n",
       "      <th>PKGen187:GT</th>\n",
       "      <th>PKGen187:DP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>634112</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "      <td>PGRA.SER.GRA_1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>204.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>735.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>138.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>290.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778639</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779047</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>24.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826893</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827209</th>\n",
       "      <th>C</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>35.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              INTERVAL_ID AS22WG009:GT  AS22WG009:DP  \\\n",
       "CHROM POS    ALT REF                                                   \n",
       "chr1  634112 C   T         PGRA.SER.GRA_1          0/1          86.0   \n",
       "      778639 G   A    PGRA.SER.GRA.PSER_2          NaN           NaN   \n",
       "      779047 A   G    PGRA.SER.GRA.PSER_2          1/1          34.0   \n",
       "      826893 A   G    PGRA.SER.GRA.PSER_3          1/1          30.0   \n",
       "      827209 C   G    PGRA.SER.GRA.PSER_3          1/1          25.0   \n",
       "\n",
       "                     AS22WG010:GT  AS22WG010:DP AS22WG011:GT  AS22WG011:DP  \\\n",
       "CHROM POS    ALT REF                                                         \n",
       "chr1  634112 C   T            0/1          46.0          0/1          46.0   \n",
       "      778639 G   A            NaN           NaN          NaN           NaN   \n",
       "      779047 A   G            1/1          20.0          1/1          21.0   \n",
       "      826893 A   G            0/1          25.0          1/1          22.0   \n",
       "      827209 C   G            0/1          35.0          1/1          18.0   \n",
       "\n",
       "                     AS22WG023:GT  AS22WG023:DP AS22WG024:GT  ...  \\\n",
       "CHROM POS    ALT REF                                          ...   \n",
       "chr1  634112 C   T            0/1          49.0          0/1  ...   \n",
       "      778639 G   A            NaN           NaN          NaN  ...   \n",
       "      779047 A   G            1/1          27.0          1/1  ...   \n",
       "      826893 A   G            1/1          17.0          1/1  ...   \n",
       "      827209 C   G            1/1          26.0          1/1  ...   \n",
       "\n",
       "                      PKGen166:GT PKGen166:DP  PKGen177:GT PKGen177:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       204.0          1/1       735.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        26.0          1/1        27.0   \n",
       "      826893 A   G            1/1        30.0          1/1        36.0   \n",
       "      827209 C   G            1/1        43.0          1/1        43.0   \n",
       "\n",
       "                      PKGen186:GT PKGen186:DP  PKGen172:GT PKGen172:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       138.0          1/1       290.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        24.0          1/1        23.0   \n",
       "      826893 A   G            1/1        36.0          1/1        26.0   \n",
       "      827209 C   G            0/1        31.0          0/1        29.0   \n",
       "\n",
       "                      PKGen187:GT PKGen187:DP  \n",
       "CHROM POS    ALT REF                           \n",
       "chr1  634112 C   T            NaN         NaN  \n",
       "      778639 G   A            NaN         NaN  \n",
       "      779047 A   G            1/1        46.0  \n",
       "      826893 A   G            1/1        36.0  \n",
       "      827209 C   G            1/1        28.0  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdf = GnomAD_df(path='data/main.prq',\n",
    "                peak_file='data/hATAC_mATAC_peak.bed.gz',\n",
    "                remove_unkown=True,\n",
    "                remove_phased_gt=True,\n",
    "                only_peak_variants=True)\n",
    "gdf.get_table().head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7683ad8a",
   "metadata": {},
   "source": [
    "### Explanation:\n",
    "The class constructor have only one mandatory variable, the path to the parquet file. <br>\n",
    "the rest of the argument, except the peak_file (it can be left empty), are the deafult argument. <br>\n",
    "Variable description:<br>\n",
    "- peak_file - bed file which contains the following columns - CHROM,FROM,TO,INTERVAL_ID,  (w/o header)\n",
    "- remove_unknown - add a filter which removes unkwon genotypes (e.g \"./.\")' \n",
    "- remove_phased_gt -  add a filter which replace phased genotype (e.g \"0|1\" -> \"0/1\")\n",
    "- only_peak_variants -  add a filter which removes variants outside of peak interval \n",
    "<br>\n",
    "\n",
    "Whenever get_table or any other function that requeires to apply the filters, only then the class creates a sub dataframe which the filters apllied on it. <br>\n",
    "During the filteration process, the filter's descriptions will be printed, unless verbos=False. <br>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ef0786e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "applying filters\n",
      "done\n",
      "getting table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009:GT</th>\n",
       "      <th>AS22WG009:DP</th>\n",
       "      <th>AS22WG010:GT</th>\n",
       "      <th>AS22WG010:DP</th>\n",
       "      <th>AS22WG011:GT</th>\n",
       "      <th>AS22WG011:DP</th>\n",
       "      <th>AS22WG023:GT</th>\n",
       "      <th>AS22WG023:DP</th>\n",
       "      <th>AS22WG024:GT</th>\n",
       "      <th>...</th>\n",
       "      <th>PKGen166:GT</th>\n",
       "      <th>PKGen166:DP</th>\n",
       "      <th>PKGen177:GT</th>\n",
       "      <th>PKGen177:DP</th>\n",
       "      <th>PKGen186:GT</th>\n",
       "      <th>PKGen186:DP</th>\n",
       "      <th>PKGen172:GT</th>\n",
       "      <th>PKGen172:DP</th>\n",
       "      <th>PKGen187:GT</th>\n",
       "      <th>PKGen187:DP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>10117</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10141</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10146</th>\n",
       "      <th>A</th>\n",
       "      <th>AC</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10151</th>\n",
       "      <th>A</th>\n",
       "      <th>T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10153</th>\n",
       "      <th>T</th>\n",
       "      <th>A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    INTERVAL_ID AS22WG009:GT  AS22WG009:DP AS22WG010:GT  \\\n",
       "CHROM POS   ALT REF                                                       \n",
       "chr1  10117 A   C           NaN          NaN           NaN          NaN   \n",
       "      10141 A   C           NaN          NaN           NaN          NaN   \n",
       "      10146 A   AC          NaN          NaN           NaN          NaN   \n",
       "      10151 A   T           NaN          NaN           NaN          NaN   \n",
       "      10153 T   A           NaN          NaN           NaN          NaN   \n",
       "\n",
       "                     AS22WG010:DP AS22WG011:GT  AS22WG011:DP AS22WG023:GT  \\\n",
       "CHROM POS   ALT REF                                                         \n",
       "chr1  10117 A   C             NaN          NaN           NaN          NaN   \n",
       "      10141 A   C             NaN          NaN           NaN          NaN   \n",
       "      10146 A   AC            NaN          NaN           NaN          NaN   \n",
       "      10151 A   T             NaN          NaN           NaN          NaN   \n",
       "      10153 T   A             NaN          NaN           NaN          NaN   \n",
       "\n",
       "                     AS22WG023:DP AS22WG024:GT  ...  PKGen166:GT PKGen166:DP  \\\n",
       "CHROM POS   ALT REF                             ...                            \n",
       "chr1  10117 A   C             NaN          NaN  ...          NaN         NaN   \n",
       "      10141 A   C             NaN          NaN  ...          NaN         NaN   \n",
       "      10146 A   AC            NaN          1/1  ...          NaN         NaN   \n",
       "      10151 A   T             NaN          NaN  ...          NaN         NaN   \n",
       "      10153 T   A             NaN          NaN  ...          NaN         NaN   \n",
       "\n",
       "                     PKGen177:GT PKGen177:DP  PKGen186:GT PKGen186:DP  \\\n",
       "CHROM POS   ALT REF                                                     \n",
       "chr1  10117 A   C            NaN         NaN          NaN         NaN   \n",
       "      10141 A   C            NaN         NaN          NaN         NaN   \n",
       "      10146 A   AC           NaN         NaN          NaN         NaN   \n",
       "      10151 A   T            NaN         NaN          NaN         NaN   \n",
       "      10153 T   A            NaN         NaN          NaN         NaN   \n",
       "\n",
       "                     PKGen172:GT PKGen172:DP  PKGen187:GT PKGen187:DP  \n",
       "CHROM POS   ALT REF                                                    \n",
       "chr1  10117 A   C            NaN         NaN          NaN         NaN  \n",
       "      10141 A   C            NaN         NaN          NaN         NaN  \n",
       "      10146 A   AC           NaN         NaN          NaN         NaN  \n",
       "      10151 A   T            NaN         NaN          NaN         NaN  \n",
       "      10153 T   A            NaN         NaN          NaN         NaN  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdf.reset_table(remove_phased_gt=False,\n",
    "    remove_unkown=False,\n",
    "    only_peak_variants=False)\n",
    "gdf.get_table().head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05f3a815",
   "metadata": {},
   "source": [
    "### Explanation:\n",
    "In the cell below, No filter was applied. The original table is shown. <br>\n",
    "Now lets filter it up!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "33f587a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "applying filters\n",
      "removing variants outside of peak interval\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "done\n",
      "getting table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009:GT</th>\n",
       "      <th>AS22WG009:DP</th>\n",
       "      <th>AS22WG010:GT</th>\n",
       "      <th>AS22WG010:DP</th>\n",
       "      <th>AS22WG011:GT</th>\n",
       "      <th>AS22WG011:DP</th>\n",
       "      <th>AS22WG023:GT</th>\n",
       "      <th>AS22WG023:DP</th>\n",
       "      <th>AS22WG024:GT</th>\n",
       "      <th>...</th>\n",
       "      <th>PKGen166:GT</th>\n",
       "      <th>PKGen166:DP</th>\n",
       "      <th>PKGen177:GT</th>\n",
       "      <th>PKGen177:DP</th>\n",
       "      <th>PKGen186:GT</th>\n",
       "      <th>PKGen186:DP</th>\n",
       "      <th>PKGen172:GT</th>\n",
       "      <th>PKGen172:DP</th>\n",
       "      <th>PKGen187:GT</th>\n",
       "      <th>PKGen187:DP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>634112</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "      <td>PGRA.SER.GRA_1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>204.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>735.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>138.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>290.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778639</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779047</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>24.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826893</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827209</th>\n",
       "      <th>C</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>35.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              INTERVAL_ID AS22WG009:GT  AS22WG009:DP  \\\n",
       "CHROM POS    ALT REF                                                   \n",
       "chr1  634112 C   T         PGRA.SER.GRA_1          0/1          86.0   \n",
       "      778639 G   A    PGRA.SER.GRA.PSER_2          NaN           NaN   \n",
       "      779047 A   G    PGRA.SER.GRA.PSER_2          1/1          34.0   \n",
       "      826893 A   G    PGRA.SER.GRA.PSER_3          1/1          30.0   \n",
       "      827209 C   G    PGRA.SER.GRA.PSER_3          1/1          25.0   \n",
       "\n",
       "                     AS22WG010:GT  AS22WG010:DP AS22WG011:GT  AS22WG011:DP  \\\n",
       "CHROM POS    ALT REF                                                         \n",
       "chr1  634112 C   T            0/1          46.0          0/1          46.0   \n",
       "      778639 G   A            NaN           NaN          NaN           NaN   \n",
       "      779047 A   G            1/1          20.0          1/1          21.0   \n",
       "      826893 A   G            0/1          25.0          1/1          22.0   \n",
       "      827209 C   G            0/1          35.0          1/1          18.0   \n",
       "\n",
       "                     AS22WG023:GT  AS22WG023:DP AS22WG024:GT  ...  \\\n",
       "CHROM POS    ALT REF                                          ...   \n",
       "chr1  634112 C   T            0/1          49.0          0/1  ...   \n",
       "      778639 G   A            NaN           NaN          NaN  ...   \n",
       "      779047 A   G            1/1          27.0          1/1  ...   \n",
       "      826893 A   G            1/1          17.0          1/1  ...   \n",
       "      827209 C   G            1/1          26.0          1/1  ...   \n",
       "\n",
       "                      PKGen166:GT PKGen166:DP  PKGen177:GT PKGen177:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       204.0          1/1       735.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        26.0          1/1        27.0   \n",
       "      826893 A   G            1/1        30.0          1/1        36.0   \n",
       "      827209 C   G            1/1        43.0          1/1        43.0   \n",
       "\n",
       "                      PKGen186:GT PKGen186:DP  PKGen172:GT PKGen172:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       138.0          1/1       290.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        24.0          1/1        23.0   \n",
       "      826893 A   G            1/1        36.0          1/1        26.0   \n",
       "      827209 C   G            0/1        31.0          0/1        29.0   \n",
       "\n",
       "                      PKGen187:GT PKGen187:DP  \n",
       "CHROM POS    ALT REF                           \n",
       "chr1  634112 C   T            NaN         NaN  \n",
       "      778639 G   A            NaN         NaN  \n",
       "      779047 A   G            1/1        46.0  \n",
       "      826893 A   G            1/1        36.0  \n",
       "      827209 C   G            1/1        28.0  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdf.remove_non_peak_variants()\n",
    "gdf.remove_phasing()\n",
    "gdf.remove_unkown()\n",
    "gdf.get_table().head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a8b8183",
   "metadata": {},
   "source": [
    "### Explanation:\n",
    "Magic! We got the same table from 2 cells ago! <br>\n",
    "Notice that the order of filteration changed. This is important, if we would like to remove a filter, we can do it by it's index. <br>\n",
    "Of course there is a simple way to know the filters index... <br>\n",
    "\n",
    "Lets see:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "dd30cf26",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "applying filters\n",
      "removing variants outside of peak interval\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "done\n",
      "getting table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009:GT</th>\n",
       "      <th>AS22WG009:DP</th>\n",
       "      <th>AS22WG010:GT</th>\n",
       "      <th>AS22WG010:DP</th>\n",
       "      <th>AS22WG011:GT</th>\n",
       "      <th>AS22WG011:DP</th>\n",
       "      <th>AS22WG023:GT</th>\n",
       "      <th>AS22WG023:DP</th>\n",
       "      <th>AS22WG024:GT</th>\n",
       "      <th>...</th>\n",
       "      <th>PKGen166:GT</th>\n",
       "      <th>PKGen166:DP</th>\n",
       "      <th>PKGen177:GT</th>\n",
       "      <th>PKGen177:DP</th>\n",
       "      <th>PKGen186:GT</th>\n",
       "      <th>PKGen186:DP</th>\n",
       "      <th>PKGen172:GT</th>\n",
       "      <th>PKGen172:DP</th>\n",
       "      <th>PKGen187:GT</th>\n",
       "      <th>PKGen187:DP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>634112</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "      <td>PGRA.SER.GRA_1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>204.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>735.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>138.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>290.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778639</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779047</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>24.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826893</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827209</th>\n",
       "      <th>C</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>35.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              INTERVAL_ID AS22WG009:GT  AS22WG009:DP  \\\n",
       "CHROM POS    ALT REF                                                   \n",
       "chr1  634112 C   T         PGRA.SER.GRA_1          0/1          86.0   \n",
       "      778639 G   A    PGRA.SER.GRA.PSER_2          NaN           NaN   \n",
       "      779047 A   G    PGRA.SER.GRA.PSER_2          1/1          34.0   \n",
       "      826893 A   G    PGRA.SER.GRA.PSER_3          1/1          30.0   \n",
       "      827209 C   G    PGRA.SER.GRA.PSER_3          1/1          25.0   \n",
       "\n",
       "                     AS22WG010:GT  AS22WG010:DP AS22WG011:GT  AS22WG011:DP  \\\n",
       "CHROM POS    ALT REF                                                         \n",
       "chr1  634112 C   T            0/1          46.0          0/1          46.0   \n",
       "      778639 G   A            NaN           NaN          NaN           NaN   \n",
       "      779047 A   G            1/1          20.0          1/1          21.0   \n",
       "      826893 A   G            0/1          25.0          1/1          22.0   \n",
       "      827209 C   G            0/1          35.0          1/1          18.0   \n",
       "\n",
       "                     AS22WG023:GT  AS22WG023:DP AS22WG024:GT  ...  \\\n",
       "CHROM POS    ALT REF                                          ...   \n",
       "chr1  634112 C   T            0/1          49.0          0/1  ...   \n",
       "      778639 G   A            NaN           NaN          NaN  ...   \n",
       "      779047 A   G            1/1          27.0          1/1  ...   \n",
       "      826893 A   G            1/1          17.0          1/1  ...   \n",
       "      827209 C   G            1/1          26.0          1/1  ...   \n",
       "\n",
       "                      PKGen166:GT PKGen166:DP  PKGen177:GT PKGen177:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       204.0          1/1       735.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        26.0          1/1        27.0   \n",
       "      826893 A   G            1/1        30.0          1/1        36.0   \n",
       "      827209 C   G            1/1        43.0          1/1        43.0   \n",
       "\n",
       "                      PKGen186:GT PKGen186:DP  PKGen172:GT PKGen172:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       138.0          1/1       290.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        24.0          1/1        23.0   \n",
       "      826893 A   G            1/1        36.0          1/1        26.0   \n",
       "      827209 C   G            0/1        31.0          0/1        29.0   \n",
       "\n",
       "                      PKGen187:GT PKGen187:DP  \n",
       "CHROM POS    ALT REF                           \n",
       "chr1  634112 C   T            NaN         NaN  \n",
       "      778639 G   A            NaN         NaN  \n",
       "      779047 A   G            1/1        46.0  \n",
       "      826893 A   G            1/1        36.0  \n",
       "      827209 C   G            1/1        28.0  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "look on AS22WG009:GT, 4th & 5th row, \n",
      " it's written NaN since we filtered according to dp thrshold of 31 and above \n",
      "\n",
      "\n",
      "applying filters\n",
      "removing variants outside of peak interval\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "Removing variants with reading depth 31 or below\n",
      "done\n",
      "getting table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009:GT</th>\n",
       "      <th>AS22WG009:DP</th>\n",
       "      <th>AS22WG010:GT</th>\n",
       "      <th>AS22WG010:DP</th>\n",
       "      <th>AS22WG011:GT</th>\n",
       "      <th>AS22WG011:DP</th>\n",
       "      <th>AS22WG023:GT</th>\n",
       "      <th>AS22WG023:DP</th>\n",
       "      <th>AS22WG024:GT</th>\n",
       "      <th>...</th>\n",
       "      <th>PKGen166:GT</th>\n",
       "      <th>PKGen166:DP</th>\n",
       "      <th>PKGen177:GT</th>\n",
       "      <th>PKGen177:DP</th>\n",
       "      <th>PKGen186:GT</th>\n",
       "      <th>PKGen186:DP</th>\n",
       "      <th>PKGen172:GT</th>\n",
       "      <th>PKGen172:DP</th>\n",
       "      <th>PKGen187:GT</th>\n",
       "      <th>PKGen187:DP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>634112</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "      <td>PGRA.SER.GRA_1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>204.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>735.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>138.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>290.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778639</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779047</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826893</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827209</th>\n",
       "      <th>C</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>35.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              INTERVAL_ID AS22WG009:GT  AS22WG009:DP  \\\n",
       "CHROM POS    ALT REF                                                   \n",
       "chr1  634112 C   T         PGRA.SER.GRA_1          0/1          86.0   \n",
       "      778639 G   A    PGRA.SER.GRA.PSER_2          NaN           NaN   \n",
       "      779047 A   G    PGRA.SER.GRA.PSER_2          1/1          34.0   \n",
       "      826893 A   G    PGRA.SER.GRA.PSER_3          NaN          30.0   \n",
       "      827209 C   G    PGRA.SER.GRA.PSER_3          NaN          25.0   \n",
       "\n",
       "                     AS22WG010:GT  AS22WG010:DP AS22WG011:GT  AS22WG011:DP  \\\n",
       "CHROM POS    ALT REF                                                         \n",
       "chr1  634112 C   T            0/1          46.0          0/1          46.0   \n",
       "      778639 G   A            NaN           NaN          NaN           NaN   \n",
       "      779047 A   G            NaN          20.0          NaN          21.0   \n",
       "      826893 A   G            NaN          25.0          NaN          22.0   \n",
       "      827209 C   G            0/1          35.0          NaN          18.0   \n",
       "\n",
       "                     AS22WG023:GT  AS22WG023:DP AS22WG024:GT  ...  \\\n",
       "CHROM POS    ALT REF                                          ...   \n",
       "chr1  634112 C   T            0/1          49.0          0/1  ...   \n",
       "      778639 G   A            NaN           NaN          NaN  ...   \n",
       "      779047 A   G            NaN          27.0          NaN  ...   \n",
       "      826893 A   G            NaN          17.0          NaN  ...   \n",
       "      827209 C   G            NaN          26.0          1/1  ...   \n",
       "\n",
       "                      PKGen166:GT PKGen166:DP  PKGen177:GT PKGen177:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       204.0          1/1       735.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            NaN        26.0          NaN        27.0   \n",
       "      826893 A   G            NaN        30.0          1/1        36.0   \n",
       "      827209 C   G            1/1        43.0          1/1        43.0   \n",
       "\n",
       "                      PKGen186:GT PKGen186:DP  PKGen172:GT PKGen172:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       138.0          1/1       290.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            NaN        24.0          NaN        23.0   \n",
       "      826893 A   G            1/1        36.0          NaN        26.0   \n",
       "      827209 C   G            NaN        31.0          NaN        29.0   \n",
       "\n",
       "                      PKGen187:GT PKGen187:DP  \n",
       "CHROM POS    ALT REF                           \n",
       "chr1  634112 C   T            NaN         NaN  \n",
       "      778639 G   A            NaN         NaN  \n",
       "      779047 A   G            1/1        46.0  \n",
       "      826893 A   G            1/1        36.0  \n",
       "      827209 C   G            NaN        28.0  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gdf.reset_table(remove_phased_gt=True,\n",
    "    remove_unkown=True,\n",
    "    only_peak_variants=True)\n",
    "display(gdf.get_table().head())\n",
    "gdf.filter_DP(31)\n",
    "print('\\n')\n",
    "print(\"look on AS22WG009:GT, 4th & 5th row, \\n it's written NaN since we filtered according to dp thrshold of 31 and above \")\n",
    "print('\\n')\n",
    "display(gdf.get_table().head())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e977ea88",
   "metadata": {},
   "source": [
    "Since our real threshold is 10, we need to fix it. <br>\n",
    "1st we need to find the index of the filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "923f2924",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "current filters:\n",
      "0. removing variants outside of peak interval\n",
      "1. replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "2. removed unkwon genotypes (e.g \"./.\")\n",
      "3. Removing variants with reading depth 31 or below\n",
      "\n",
      " lets remove filter no 3\n",
      "\n",
      " current filters: \n",
      "applying filters\n",
      "removing variants outside of peak interval\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "done\n",
      "getting table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009:GT</th>\n",
       "      <th>AS22WG009:DP</th>\n",
       "      <th>AS22WG010:GT</th>\n",
       "      <th>AS22WG010:DP</th>\n",
       "      <th>AS22WG011:GT</th>\n",
       "      <th>AS22WG011:DP</th>\n",
       "      <th>AS22WG023:GT</th>\n",
       "      <th>AS22WG023:DP</th>\n",
       "      <th>AS22WG024:GT</th>\n",
       "      <th>...</th>\n",
       "      <th>PKGen166:GT</th>\n",
       "      <th>PKGen166:DP</th>\n",
       "      <th>PKGen177:GT</th>\n",
       "      <th>PKGen177:DP</th>\n",
       "      <th>PKGen186:GT</th>\n",
       "      <th>PKGen186:DP</th>\n",
       "      <th>PKGen172:GT</th>\n",
       "      <th>PKGen172:DP</th>\n",
       "      <th>PKGen187:GT</th>\n",
       "      <th>PKGen187:DP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>634112</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "      <td>PGRA.SER.GRA_1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>86.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>46.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>49.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>204.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>735.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>138.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>290.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778639</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779047</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>1/1</td>\n",
       "      <td>34.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>20.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>24.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>23.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826893</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>30.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827209</th>\n",
       "      <th>C</th>\n",
       "      <th>G</th>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>1/1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>35.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>26.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>43.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1/1</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              INTERVAL_ID AS22WG009:GT  AS22WG009:DP  \\\n",
       "CHROM POS    ALT REF                                                   \n",
       "chr1  634112 C   T         PGRA.SER.GRA_1          0/1          86.0   \n",
       "      778639 G   A    PGRA.SER.GRA.PSER_2          NaN           NaN   \n",
       "      779047 A   G    PGRA.SER.GRA.PSER_2          1/1          34.0   \n",
       "      826893 A   G    PGRA.SER.GRA.PSER_3          1/1          30.0   \n",
       "      827209 C   G    PGRA.SER.GRA.PSER_3          1/1          25.0   \n",
       "\n",
       "                     AS22WG010:GT  AS22WG010:DP AS22WG011:GT  AS22WG011:DP  \\\n",
       "CHROM POS    ALT REF                                                         \n",
       "chr1  634112 C   T            0/1          46.0          0/1          46.0   \n",
       "      778639 G   A            NaN           NaN          NaN           NaN   \n",
       "      779047 A   G            1/1          20.0          1/1          21.0   \n",
       "      826893 A   G            0/1          25.0          1/1          22.0   \n",
       "      827209 C   G            0/1          35.0          1/1          18.0   \n",
       "\n",
       "                     AS22WG023:GT  AS22WG023:DP AS22WG024:GT  ...  \\\n",
       "CHROM POS    ALT REF                                          ...   \n",
       "chr1  634112 C   T            0/1          49.0          0/1  ...   \n",
       "      778639 G   A            NaN           NaN          NaN  ...   \n",
       "      779047 A   G            1/1          27.0          1/1  ...   \n",
       "      826893 A   G            1/1          17.0          1/1  ...   \n",
       "      827209 C   G            1/1          26.0          1/1  ...   \n",
       "\n",
       "                      PKGen166:GT PKGen166:DP  PKGen177:GT PKGen177:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       204.0          1/1       735.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        26.0          1/1        27.0   \n",
       "      826893 A   G            1/1        30.0          1/1        36.0   \n",
       "      827209 C   G            1/1        43.0          1/1        43.0   \n",
       "\n",
       "                      PKGen186:GT PKGen186:DP  PKGen172:GT PKGen172:DP  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T            0/1       138.0          1/1       290.0   \n",
       "      778639 G   A            NaN         NaN          NaN         NaN   \n",
       "      779047 A   G            1/1        24.0          1/1        23.0   \n",
       "      826893 A   G            1/1        36.0          1/1        26.0   \n",
       "      827209 C   G            0/1        31.0          0/1        29.0   \n",
       "\n",
       "                      PKGen187:GT PKGen187:DP  \n",
       "CHROM POS    ALT REF                           \n",
       "chr1  634112 C   T            NaN         NaN  \n",
       "      778639 G   A            NaN         NaN  \n",
       "      779047 A   G            1/1        46.0  \n",
       "      826893 A   G            1/1        36.0  \n",
       "      827209 C   G            1/1        28.0  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(\"current filters:\")\n",
    "gdf.print_filters()\n",
    "print(\"\\n lets remove filter no 3\")\n",
    "gdf.remove_filter(3)\n",
    "print('\\n current filters: ')\n",
    "gdf.get_table().head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "378a2685",
   "metadata": {},
   "source": [
    "Yay :) the table looks similar to the original! <br>\n",
    "<br>\n",
    "## Example : Filtering the table according to thresholds\n",
    "\n",
    "After the short torturial above, see how easily is to achive a table which contains variants acoording to given thresholds.<br>\n",
    "Let's say we want only variants with reading depth (DP) > 10 & gnomAD allele frequency < 0.01 <br>\n",
    "First, lts see how many variant each sample have before applying the filters:    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0b8aa618",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "first step: getting a boolean table, which suggest if a sample have a variant or not\n",
      "\n",
      "removing variants outside of peak interval\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "done\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>AF</th>\n",
       "      <th>INTERVAL_ID</th>\n",
       "      <th>AS22WG009</th>\n",
       "      <th>AS22WG010</th>\n",
       "      <th>AS22WG011</th>\n",
       "      <th>AS22WG023</th>\n",
       "      <th>AS22WG024</th>\n",
       "      <th>AS22WG004</th>\n",
       "      <th>AS22WG005</th>\n",
       "      <th>AS22WG006</th>\n",
       "      <th>...</th>\n",
       "      <th>PKGen169</th>\n",
       "      <th>PKGen176</th>\n",
       "      <th>PKGen171</th>\n",
       "      <th>PKGen185</th>\n",
       "      <th>PKGen182</th>\n",
       "      <th>PKGen166</th>\n",
       "      <th>PKGen177</th>\n",
       "      <th>PKGen186</th>\n",
       "      <th>PKGen172</th>\n",
       "      <th>PKGen187</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>634112</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "      <td>0.581946</td>\n",
       "      <td>PGRA.SER.GRA_1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>778639</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <td>0.127933</td>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779047</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>0.819806</td>\n",
       "      <td>PGRA.SER.GRA.PSER_2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826893</th>\n",
       "      <th>A</th>\n",
       "      <th>G</th>\n",
       "      <td>0.730133</td>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827209</th>\n",
       "      <th>C</th>\n",
       "      <th>G</th>\n",
       "      <td>0.739536</td>\n",
       "      <td>PGRA.SER.GRA.PSER_3</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            AF          INTERVAL_ID  AS22WG009  AS22WG010  \\\n",
       "CHROM POS    ALT REF                                                        \n",
       "chr1  634112 C   T    0.581946       PGRA.SER.GRA_1       True       True   \n",
       "      778639 G   A    0.127933  PGRA.SER.GRA.PSER_2      False      False   \n",
       "      779047 A   G    0.819806  PGRA.SER.GRA.PSER_2       True       True   \n",
       "      826893 A   G    0.730133  PGRA.SER.GRA.PSER_3       True       True   \n",
       "      827209 C   G    0.739536  PGRA.SER.GRA.PSER_3       True       True   \n",
       "\n",
       "                      AS22WG011  AS22WG023  AS22WG024  AS22WG004  AS22WG005  \\\n",
       "CHROM POS    ALT REF                                                          \n",
       "chr1  634112 C   T         True       True       True       True       True   \n",
       "      778639 G   A        False      False      False      False      False   \n",
       "      779047 A   G         True       True       True       True       True   \n",
       "      826893 A   G         True       True       True       True       True   \n",
       "      827209 C   G         True       True       True       True       True   \n",
       "\n",
       "                      AS22WG006  ...  PKGen169  PKGen176  PKGen171  PKGen185  \\\n",
       "CHROM POS    ALT REF             ...                                           \n",
       "chr1  634112 C   T         True  ...      True      True      True      True   \n",
       "      778639 G   A        False  ...     False     False     False      True   \n",
       "      779047 A   G         True  ...      True      True      True      True   \n",
       "      826893 A   G         True  ...      True      True      True     False   \n",
       "      827209 C   G         True  ...      True      True      True     False   \n",
       "\n",
       "                      PKGen182  PKGen166  PKGen177  PKGen186  PKGen172  \\\n",
       "CHROM POS    ALT REF                                                     \n",
       "chr1  634112 C   T        True      True      True      True      True   \n",
       "      778639 G   A       False     False     False     False     False   \n",
       "      779047 A   G        True      True      True      True      True   \n",
       "      826893 A   G        True      True      True      True      True   \n",
       "      827209 C   G        True      True      True      True      True   \n",
       "\n",
       "                      PKGen187  \n",
       "CHROM POS    ALT REF            \n",
       "chr1  634112 C   T       False  \n",
       "      778639 G   A       False  \n",
       "      779047 A   G        True  \n",
       "      826893 A   G        True  \n",
       "      827209 C   G        True  \n",
       "\n",
       "[5 rows x 39 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now lets count how many variants the first 5 sample have -\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AS22WG009    39517\n",
       "AS22WG010    36498\n",
       "AS22WG011    36302\n",
       "AS22WG023    36770\n",
       "AS22WG024    36837\n",
       "dtype: int64"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gdf.reset_table() # make sure we cleaned all the mess from the cell above\n",
    "\n",
    "print('first step: getting a boolean table, which suggest if a sample have a variant or not\\n')\n",
    "bool_df = gdf.bool_variant_df()\n",
    "display(bool_df.head())\n",
    "print('\\nNow lets count how many variants the first 5 sample have -')\n",
    "all_v = bool_df.drop(columns=['AF','INTERVAL_ID']).sum().head()\n",
    "display(all_v)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec4f76be",
   "metadata": {},
   "source": [
    "Lets apply the filters:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "96e36ec8",
   "metadata": {},
   "outputs": [],
   "source": [
    "DP_t = 10 \n",
    "AF_t = 0.01\n",
    "gdf.filter_DP(DP_t)\n",
    "gdf.filter_AF(AF_t, remove_unkwon=True)\n",
    "bool_df = gdf.bool_variant_df()\n",
    "print('\\nNow lets count how many variants the first 5 sample have -')\n",
    "filter_v = bool_df.drop(columns=['AF','INTERVAL_ID']).sum().head()\n",
    "display(pd.concat([all_v.rename('total_variants'),filter_v.rename(\"DP > 10 & AF < 0.01\")], axis=1))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f8fd3f8",
   "metadata": {},
   "source": [
    "Notice the  remove_unkwon variable in 'filter_AF' function. If set to 'True' (default is 'False'), variants without reported allele frequency in gnomAD will be removed. <br>\n",
    "\n",
    "Since I love one liner, pipping is also possible :) <br>\n",
    "Let's do the same with less lines of code! <br>\n",
    "(Note: The following is just POC, this is a bad example of code writing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fd5ab738",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "removing variants outside of peak interval\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "Removing variants with reading depth 10 or below\n",
      "Removing variants with allele frequency above 0.01 and variants with no record on gnomAD\n",
      "done\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>total_variants</th>\n",
       "      <th>DP &gt; 10 &amp; AF &lt; 0.01</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AS22WG009</th>\n",
       "      <td>39517</td>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AS22WG010</th>\n",
       "      <td>36498</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AS22WG011</th>\n",
       "      <td>36302</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AS22WG023</th>\n",
       "      <td>36770</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AS22WG024</th>\n",
       "      <td>36837</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           total_variants  DP > 10 & AF < 0.01\n",
       "AS22WG009           39517                  179\n",
       "AS22WG010           36498                  103\n",
       "AS22WG011           36302                   91\n",
       "AS22WG023           36770                   30\n",
       "AS22WG024           36837                   42"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gdf.reset_table()\n",
    "filter_v = gdf.filter_DP(DP_t).filter_AF(AF_t,  remove_unkwon=True).bool_variant_df().drop(columns=['AF','INTERVAL_ID']).sum().head()\n",
    "display(pd.concat([all_v.rename('total_variants'),filter_v.rename(\"DP > 10 & AF < 0.01\")], axis=1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93beac5f",
   "metadata": {},
   "source": [
    "Since the function of getting the amount of variant in each sample is something we want to do often, there is a function that does that. <br>\n",
    "so to summrise it all, a general pipline is:\n",
    "1. load the parquet file into the class object (remember to adjust default filters if needed)\n",
    "2. apply the filters  - AF,DP & some more filters avaiable\n",
    "3. get the amount of varints in each sample (or other data :) )\n",
    "<br>\n",
    "\n",
    "## Final example - Getting the amount of variant in each samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0ee8d283",
   "metadata": {},
   "outputs": [],
   "source": [
    "gdf = GnomAD_df(path='data/main.prq',\n",
    "                peak_file='data/hATAC_mATAC_peak.bed.gz',\n",
    "                remove_unkown=True,\n",
    "                remove_phased_gt=True,\n",
    "                only_peak_variants=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e1240ea4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing variants with allele frequency above 0.01 already applied\n",
      "removing variants outside of peak interval\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "Removing variants with allele frequency above 0.01\n",
      "Removing variants with reading depth 10 or below\n",
      "done\n",
      "the amount of variants each sample have in the hATAC interval - \n"
     ]
    }
   ],
   "source": [
    "# # getting the total amount of variants for each sample\n",
    "all_v = gdf.count_sample_variants(verbos=True).rename('all_variants')\n",
    "\n",
    "# # getting the amount of variants only in ATAC intervals\n",
    "only_interval = gdf.remove_non_peak_variants().count_sample_variants(verbos=True).rename('interval_variants')\n",
    "\n",
    "# # getting the amount of variants in interval & AF < 0.01\n",
    "af_v = gdf.filter_AF(AF_t).count_sample_variants(verbos=True).rename('interval_variants_AF<0.01')\n",
    "\n",
    "# # getting the amount of variants in interval & DP > 10\n",
    "dp_v = gdf.remove_filter(-1).filter_DP(DP_t).count_sample_variants(verbos=True).rename('interval_variants_DP>10') \n",
    "\n",
    "#\n",
    "both_v =  gdf.filter_AF(AF_t).count_sample_variants(verbos=True).rename('interval_variants_AF<0.01_DP>10')\n",
    "print(\"the amount of variants each sample have in the hATAC interval - \")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6f60be71",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'all_v' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_1293/38620425.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcount_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mall_v\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0monly_interval\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0maf_v\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdp_v\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mboth_v\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0morder\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m\"AS22WG001\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG002\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG003\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG004\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG005\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG006\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG007\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG008\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG009\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG010\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG011\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG012\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG013\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG014\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG015\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG016\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG017\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG018\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG019\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG020\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG021\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG022\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG023\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"AS22WG024\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcount_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcount_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcount_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mcount_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstartswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'P'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mdisplay\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcount_df\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'all_v' is not defined"
     ]
    }
   ],
   "source": [
    "count_df = pd.concat([all_v, only_interval,af_v, dp_v,both_v], axis=1)\n",
    "order = [\"AS22WG001\",\"AS22WG002\",\"AS22WG003\",\"AS22WG004\",\"AS22WG005\",\"AS22WG006\",\"AS22WG007\",\"AS22WG008\",\"AS22WG009\",\"AS22WG010\",\"AS22WG011\",\"AS22WG012\",\"AS22WG013\",\"AS22WG014\",\"AS22WG015\",\"AS22WG016\",\"AS22WG017\",\"AS22WG018\",\"AS22WG019\",\"AS22WG020\",\"AS22WG021\",\"AS22WG022\",\"AS22WG023\",\"AS22WG024\"]\n",
    "count_df = count_df.iloc[[count_df.index.tolist().index(i) for i in order] + [i for i in count_df.index.tolist() if i.startswith('P')]]\n",
    "display(count_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "338847fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAWsAAAE4CAYAAACdXNIhAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAfeklEQVR4nO3debwkZX3v8c+PmUEQGGCYwyLDOIb1EhIYOAIKBkEiA3hVFFQMXBWSCTduqEkENUF9obhEFoNw70RkkbDKIhpAUAYRkGUWYBgWQbYwcmUAgSHXGJZf/niew6mp6T719Ex1Tz11vu/Xq16nu/p76vl1d/Wvq6uru83dERGRZltjdRcgIiLV1KxFRDKgZi0ikgE1axGRDKhZi4hkQM1aRCQDfWvWZvY9M3vSzO5OzL/PzO4xs8Vmdl6/6hIRyZH16zhrM/sz4AXgHHffoSK7NXARsI+7/87MNnb3J/tSmIhIhvq2Ze3uNwDPFOeZ2ZZmdrWZzTezX5jZdvGivwK+4+6/i/+rRi0iUjDofdZzgI+7+y7A3wKnxfnbANuY2U1mdouZzRpwXSIijTZxUAOZ2brAm4GLzWxk9msKdWwNvBWYBtxgZn/i7s8Oqj4RkSYbWLMmbMU/6+47dbjsceBWd38ReNjMfkVo3rcPsD4RkcYa2G4Qd3+e0IgPAbBgx3jx5YStasxsKmG3yEODqk1EpOn6eeje+cAvgW3N7HEzOxL4C+BIM7sTWAy8K8Z/AjxtZvcAc4G/c/en+1WbiEhu+nbonoiI1EefYBQRyYCatYhIBvpyNMjUqVN9xowZ/Vi0iEgrzZ8//yl3H+p2eV+a9YwZM5g3b14/Fi0i0kpm9uhYl2s3iIhIBtSsRUQyoGYtIpIBNWsRkQyoWYuIZEDNWkQkA2rWIiIZULMWEcnAIL/PWmQ5M475t+XOP/K1A1dTJSLNp2YtfVFXI1ZDFwm0G0REJANJW9Zm9giwDHgZeMndh/tZlDRXeUsXtLUrMgi97AbZ292f6lslDTHIZpQylpqjiID2WYs0lvbXS1HqPmsHrjGz+WY2u58FiYjIilK3rPd09yVmtjFwrZnd5+43FAOxic8GmD59es1lNot2TYjIoCU1a3dfEv8+aWaXAbsCN5Qyc4A5AMPDw+P+V3jV0Mcv7b6Qfqhs1ma2DrCGuy+Lp98OfLnvlYm0mBq69Cply3oT4DIzG8mf5+5X97UqkYZSk5XVpbJZu/tDwI4DqEVEVhPttms+Hbon0nJqxO2gj5uLiGRAzVpEJANq1iIiGVCzFhHJgJq1iEgG1KxFRDKgZi0ikgE1axGRDKhZi4hkQM1aRCQDatYiIhlQsxYRyYCatYhIBtSsRUQyoGYtIpIBNWsRkQyoWYuIZEDNWkQkA2rWIiIZULMWEcmAfjBXRGqjH+ftH21Zi4hkQFvWIjJQ2vpeOdqyFhHJgJq1iEgGtBtERJI0bffFIOtJHaucq7MeNWsRaZymPTE0gXaDiIhkQM1aRCQD2g0iIjJAK7tfO7lZm9kEYB6wxN3f0UtxIiKrQ5v2ffeyZf1J4F5gcp9qEREZuFwaetI+azObBhwIfLe/5YiISCepbzCeDPw98Er/ShERkW4qm7WZvQN40t3nV+Rmm9k8M5u3dOnS2goUEZG0Les9gHea2SPABcA+ZnZuOeTuc9x92N2Hh4aGai5TRGR8q2zW7n6su09z9xnAB4Dr3P2wvlcmIiKv0odiREQy0NOHYtz9euD6vlQiIiJdactaRCQDatYiIhlQsxYRyYCatYhIBtSsRUQyoGYtIpIBNWsRkQyoWYuIZEDNWkQkA2rWIiIZULMWEcmAmrWISAbUrEVEMqBmLSKSATVrEZEMqFmLiGRAzVpEJANq1iIiGVCzFhHJgJq1iEgG1KxFRDKgZi0ikgE1axGRDKhZi4hkQM1aRCQDatYiIhlQsxYRyYCatYhIBtSsRUQyoGYtIpIBNWsRkQxUNmszW8vMbjOzO81ssZl9aRCFiYjIqIkJmT8A+7j7C2Y2CbjRzK5y91v6XJuIiESVzdrdHXghnp0UJ+9nUSIisrykfdZmNsHM7gCeBK5191v7WpWIiCwnqVm7+8vuvhMwDdjVzHYoZ8xstpnNM7N5S5curblMEZHxraejQdz9WWAuMKvDZXPcfdjdh4eGhmoqT0REIO1okCEz2yCeXhv4c+C+PtclIiIFKUeDbAacbWYTCM39Inf/cX/LEhGRopSjQe4CZg6gFhER6UKfYBQRyYCatYhIBtSsRUQyoGYtIpIBNWsRkQykHLqXhRnH/NsK8x752oGroRKRwdF6P360plnL+FVuWGpW0kZ9bdZ6EA2Gtq5E2m9cbVmrqYlIrvQGo4hIBtSsRUQyoGYtIpIBNWsRkQyoWYuIZEDNWkQkA2rWIiIZULMWEcmAmrWISAbUrEVEMqBmLSKSATVrEZEMqFmLiGRAzVpEJANq1iIiGVCzFhHJgJq1iEgG1KxFRDKgZi0ikgE1axGRDKhZi4hkYLX/unn5F8c7/dp4SkbGlvLL7m3+9XetZ5K7yi1rM9vCzOaa2T1mttjMPjmIwkREZFTKlvVLwGfcfYGZrQfMN7Nr3f2ePtcmIiJR5Za1uz/h7gvi6WXAvcDm/S5MRERG9fQGo5nNAGYCt/alGhER6Si5WZvZusAlwNHu/nyHy2eb2Twzm7d06dI6axQRGfeSmrWZTSI06n9190s7Zdx9jrsPu/vw0NBQnTWKiIx7KUeDGHAGcK+7n9j/kkREpCxly3oP4HBgHzO7I04H9LkuEREpqDx0z91vBGwAtYiISBf6uLmISAbUrEVEMqBmLSKSATVrEZEMqFmLiGRAzVpEJANq1iIiGVCzFhHJgJq1iEgG1KxFRDKgZi0ikgE1axGRDKhZi4hkQM1aRCQDatYiIhlQsxYRyYCatYhIBtSsRUQyoGYtIpIBNWsRkQyoWYuIZEDNWkQkA2rWIiIZULMWEcmAmrWISAbUrEVEMqBmLSKSATVrEZEMqFmLiGRAzVpEJANq1iIiGahs1mb2PTN70szuHkRBIiKyopQt67OAWX2uQ0RExlDZrN39BuCZAdQiIiJdaJ+1iEgGamvWZjbbzOaZ2bylS5fWtVgREaHGZu3uc9x92N2Hh4aG6lqsiIig3SAiIllIOXTvfOCXwLZm9riZHdn/skREpGhiVcDdDx1EISIi0p12g4iIZEDNWkQkA2rWIiIZULMWEcmAmrWISAbUrEVEMqBmLSKSATVrEZEMqFmLiGRAzVpEJANq1iIiGVCzFhHJgJq1iEgG1KxFRDKgZi0ikgE1axGRDKhZi4hkQM1aRCQDatYiIhlQsxYRyYCatYhIBtSsRUQyoGYtIpIBNWsRkQyoWYuIZEDNWkQkA2rWIiIZULMWEcmAmrWISAbUrEVEMqBmLSKSATVrEZEMJDVrM5tlZveb2YNmdky/ixIRkeVVNmszmwB8B9gf2B441My273dhIiIyKmXLelfgQXd/yN3/C7gAeFd/yxIRkSJz97EDZgcDs9z9L+P5w4Hd3P1jpdxsYHY8uy1wf+HiqcBTFbUoU61pNSmjzKpkmljT6sy83t2HuqbdfcwJOBj4buH84cCpVf9XWsY8ZVYt08SalFFmVTJNrKlpmeKUshtkCbBF4fy0OE9ERAYkpVnfDmxtZm8wszWBDwBX9LcsEREpmlgVcPeXzOxjwE+ACcD33H1xj+PMUWaVM4MeTxll+p0Z9Hg5Zl5V+QajiIisfvoEo4hIBtSsRUQyoGYtIpIBNWuRljOzKWY2JTG7c7/ryZ2ZbbQ6xq29WZvZH5nZ35rZKWZ2opkdZWaTS5n9zOx0M7siTqeb2azE5f9j/Gtm9j4zOySefpuZfdvM/sbMul4vM7suYYxzKi7f08w+bWZvL83fbeS6mtnaZvYlM/uRmX3dzNaP8z9hZlt0Wm5hOWua2f8ys33j+Q+a2alm9lEzm5RQ/0cKp7eLt826pcyswuldzeyN8fT28bodUMrvZ2ZHmtmM0vwj4t9a7g8zm1o6f1hczmwzszjvoJHmY2ZDZnaOmS0yswvNbFqcf6KZ7VFxO00xs380s7+MNX/ezH5sZt80sw0Lub3j7f9DM7vUzL5mZlt1uH1WaZ0uLKfr7ZzKzKab2QVmthS4FbjNzJ6M82bEzM6laRfgCjObaR2atpltaKXHcpexJ5vZLsXbsIe6i+vl+mZ2hpndZWbnmdkmheWfYGbfN7MPlv7/tDprivf11Hh62MweAm41s0fNbK8elrPAzL5gZlv2Mv5yevkETcIncj4BXAN8AbiZ8AVQXwHuAd4aMycDVxKO194zTh+I805JGOOx+Pc04AeEY77PBS4mfLrygpHlAHeVpkXAH0bOx8wVpelHwAsj52PmtsL4fwXcARwH3AQcU7hsMTAxnp4Tr+ueMXtpnP8c8BvgF8DfAEMdruO/AhfGWr4PXBav21nA2T3cRp8gfOz/cuAR4F2FzIL49zjgFmAecAJwHfAPwA3A52Pmq/H8ycCvgY93WE5d98eCwrK/QDhk9ENxeSfF+fcUMhcCnyJ8WOvDwLVx/tJ4nR4FvgHM7HA7XQl8HTgduB74Z+AtwJeBH8bMCcCZwGHx+n0zrgMLgUNqXqcrb+d4eot4u/4C+BwwqXDZ5fHvL4H3AxMKl02Idd0Sz79CeJzOLUy/j3+vi5nXAecQ1tuXgcfi9MWRceP9PTWe3i9e/tN42x+S2DsWdbie3wWOB14f7+OR63YJ8DXg3YT17RLgNaX1sbIm4Jk4xtuIR8Z1qyuengu8MZ7ehvgJxMT742Hgn2Idt8Xr87qe+msv4ZQbfGTlAF4LXB9PTwcWxtO/6vK/BjwQTz/fZVoGvFS6cycBTwNrxvMTWb4RnwtsF+/wGcC/x9OvH7lzY+atwF7x7xPx9F4xs7BQ5+3EBgusU7oz7+304Irn7xhZFuEVzduBMwhN5WpCQ1pvpKkVrstvC7epFS4rN77lGmDh/lg3np5BaF6fLF6nkfss3l/PA5Pj/LULYy1i9EloA0ITOqm8nJruj+JtvQBYp7DckTHuL2Tmd7udCw+qfyA8kd5HeHLappQ1YEmX5RTv34nATfH0hsDdda/TVbdzPH0tcBSwE+EJ5mZgo9L1fmCMx+lITe8Ffg7sX7js4VL2OkY3tN4DnERY748H5nS4jW4GZsTTU4E7C5e9p8v0XmBp+XEzch90uE/K8z9P2HDaiNFmXVkTYUPmY/F/lwCnALuXln1v4T65pXTZyPqYcn8Ur9dbCBs3/4/wBDC723213HgpodQprmwjz3AbUvjsO6Mr9l3EZ6fS/+5auPKPAZt0GePfO6y8V3e6U+PpgwhbK++M5x8qZdcgPMtdC+zUJXNnvD4bUfo8f6mOi4GPxNNnAsPx9DbA7eU7LZ6fBLwTOL+wwt4NrBnHXAZMifPXIj4hEJr4TsRGV5hmAL+JmcWlsdYlPDGcSKmplU+XHhz3luZPIDzRXDwyRo33x33ATGAXCg/0Uj3/l7D1uzbwLeCgOH9v4Oedbuc4708JW8oPFtbFDQkbE88x+oDeiLj1Hu/7kdt/OoUHbOG617VOV97O5dsznj+M8GS0JaPN6gJCQ9iNsHX8unj6NOCi0jpxUhxjeof7o3wfzC+cvm/kdmD0Sf5GYI3ybRRPv0h4dXhmh2lZzDwOfBr4DPAQhS1eRp/07y2OEed9ONbxaGpNLN9ApwN/T9hAeAj4apz/ccLegn0IryZOIWzIfQn4fg/3R6f1cQIwCziz03qxQj4llDoBn4wr7r8QHnQjjWsIuCGe3pmwD+2eeCNcE2/8W4BdYuZ4YNcuY3w9/r2KuNVYunxTCrst4rx1CA3qh8DjXZY7La6w3yG+LC1c9ki8Ax+OfzcrrOjFRrR+XBl/Ha/jizH/c2DHmFk4xu332vj3U/H/HiXsyvhZvE0XAcfFzBnAnl2Wc178ex3xCahw2UTCy9qX4/lbC+OuUbouIyvaj4mvMkrLOh54pc77g+Vfks8t3NavPlESnuC+yOhL8lcIT2rnAdOrbufCWIcSnvR+S9i6+ynhSXsJcWuHsCvh0Tj/MeDAwjo9cjvXtU5X3s7x/GJgrVJmX+BB4Il4fk3gfxOenBfF6WrCrrfXdBhjZry9l5bm/5TQfDYnNK5L4nwjvqIA3gfMB44g7Fa6hPBK8SzgW4VlzQd26HIbjDxhHVeaRl7FbgqcE09/A9i3wzJmMfqqobKmbusI4ZXfcYXzexN2ty2Mt+NVwF8zuhso5f64oGp9rFxfV3UBHa7oHxO+qW+7itymhK2nXYBNaxx/HWDjLpftCBxV8f8HEp9VE8Z6LfCGDvMnx7F2obQ1RXwJnrDs1xH3aRFeEh9Mlwf7GMuY1u22BfaIf1d44Mb5U4E/iafXBtbuktu8n/dHITuB+KRSmr8+8SVnaf4KTxxjLHfkZe5EYJj4BFHITInzN6hY1iqt06m3M+HJfK8OmZnEffYrMxEa8HqledOBiwiv9s5l+SfP9xZyWxGa4mWE91pOB/YrLestxCfTDmMPr2zdY1yfMWsCTqxpnL7cH+Wp9o+bx3fsdyU8E0PYSrnNEwYys+3c/b5eMmY2yd1fLGWmuvtTqZmUmlOvV13Lalqmm5W5z9qaqWNdTM1UMbODCLuEnjGzIcKbWzsTtv4/4+6Pp2RSx6tLPMrnY4Q34c8gvGH3JsIrla+6++9SMn2oZwnwPeBY4M29jlVHzbUeumfhULYHCC9RD4jTl4AHrHSYWxfXpGbi4VSPA0+Y2TWlQ516yVTWnHq96lpW0zIVku+ztmZqXBcrMzF3kFUcugh8xd2fiadPJRzBtD/hJfyZqZmUsSzhMMkecucSXo3tQtgtsylh6/j3hF0YSZkOdZ+9knWPjDUcx9qsw1h1Xa8x1bplbWb3Et5ZfqQ0/w3Ale7+P8zs293+HfiQu09OzNwOfNjdF1v4NZsTgMPd/RYzW+juMxMzKTVXZnq4/jlm6rrP2pqpa12szACY2T3uvn08fSFh3/jFhP2kf+Huf25m97v7tjEz3913ebVwszvcfafETMpYSwn79YcI+3bPd/eFK9xgCbnCuEZ4P2PzDpelZGqpO3GsWpZTvr3KKr8itUcTCe/mli0hvCkE8BHCO71/6JA7tIfMmh6/qtXdfxCbzqVm9lnAe8ik1JySqXNZTcvUdZ+1NVPXupiSgbCffcRW7v7+ePosMzs6nr7ezL5MaPjXm9lB7n6Zme1NOPIlNZMy1uPuPmxm2xDekD3Xwg9tn09oXL/qIbeGhQ+urAesa2Yz3P0RC58aXDMuJyVTV90pY9W1nLF5vTv0jyW8Y/pZ4INx+mycd2zMXAe8ucv/P9xDZh6lN3EIb6jdwehhQCmZlJorM3Uuq4GZuu6ztmbqWhcrM3FeyqGLKUfMpGRqOUwyNUfaETopmboO70wZq5blVE39eINxe8Jxw8U3q65w93vi5VOA/3T3/z/GMlIy+xIOM7qzNH8D4KPu/pWUTErNqZk6l9WkTI33WVsztayLPayvkwgfBDkiRqYB/0E44uEYd3+s9P/rE454eXqM69AxkzJWcRfNWHrITSDson3JzCYSPk+wxN2fSM3UXHfVWLVdrzH/v+5mXShsCoCPvoExbjJNrEmZ9mVirrIRd/iflToaZoyGvq67v5AwblKul5pSMqtad9VYg7heQO27QaYTPjn1JOEIgwfj6QsY/XTYSGZpwzIpNXfN1LmsBmeadp+Nu0zCY3DMzzfEzGM1ZVLGqszkWneNY1Uux91rb9YpXx7TykwTa1KmfZnUBz7w7S7TPwPPp2ZWtcmkZJpcd11jrepy3GveZ21mD7j71mNd1tbMeL/+ygx0PUs5nHAZ3Y9i+Za7T03M1HJ4Y6511zhW5XK6LONVdR+6N9/C98meTfg2NQhfH/ghwtEFbc40sSZl2peBtMMJbyd8edrN5YCZfbGHTF2HN+Zad11jpSxnbCmb36kTCV8e09ZME2tSpn2ZmEs5nHAKHb5LpZRNydRyeGOuddc4VuVyqqa+HQ0iIv1hCYcTDnKs1HpyrbuOsepQ9z7rPwa2dPcr4vmTCN+KBnCquy9oa2a8X39lBree9cLCd1Z8kfBd5xMJ+1Hd3f+ol8ygNa3uusZapeWsymZ5h03+H1F4OUD49q73En7e6fI2Z5pYkzLty5Qeb3sQPgX3Kwrft17K3Ef4cqaNCV9ruhGlr5RNzKSMVZnJte4ax6pcTtf+mhJKnVjxV1SKh7Xd2OZME2tSpn2Z0uUpDeTW8v91WE5KppbmmWvdNY5VuZxuU91Hg6xXPOPuuxfObtzyTBNrUqZ9maLn3P2qDvOL5prZN4FLKRyt4MvvUknJpIyVksm17rrGSllOR3U369+Y2W7ufmtxppntTvjS7TZnmliTMu3LFKU88HeLf4cL85zwm4K9ZOpqnrnWXddYKcvpqO43GHclfJ/rWYQfnoTwZdsfAt7v7re1NTPer78yg1vPRpjZXFbk7l75wO9Vylip9eRadx1jrYp+fOveJsBHCb/FCOHHJL/j7r9te6aJNSnTvkwv4vK+Svg9z/0tfMPim9z9jF4yg9a0uusaa5WW4yu5s7vTBLybLj+O2vZME2tSpn2ZUn4Twu/5XRXPbw8cWcpcRfil7zvj+YnAopXIpIxVmcm17hrHqlxO1/s7dcVIXHl+QPgu5AcIH5mdTemn59uaaWJNyrQvsxIN5Pb4d2Fh3h0rkamleeZad41jVS6n6/2dEup1At5A+MWRU4H5hK98vHI8ZJpYkzLty/TQQK4nHEK2IJ7fnfhLKT1mammeudZd41iVy+k21X00CADu/rCZvYbwkzprA2vFv63PNLEmZdqXif7Dwm/4Obx61MhzpcyngSuALc3sJsKPuh68EpmUsVIyudZd11gpy+mo7qNBPge8KRZwP+EXhW8B7nL3l9ucaWJNyrQvU2RmOxO+D3kH4O74fwe7+12l3ERgW8JHm+939xc7LGvMTMpYPdSTa92rPFbq9eqk7mZ9H6O/c3Yz4dM6z42HTBNrUqZ9mbJuD3wzmwxs4u4PxPOHMLp1/hN3/21KJmWsXjM51V3XWL0up+NtVmezjkVMAd4cp92BdYE7gZvd/cw2Z5pYkzKtzKQ0tDnxf86KmQeBK4HXAi+5+1GJmdqaZ4511zhW5XKo4gk7tldmIrwTuhvwd4Tfknt5vGSaWJMy7ckAc4APF84/SPiJqO8C/yfOW0jcGBs5Xzh9Yw+ZlLEqM7nWXeNYlcupmmp9g9HM3knYItiDcFD/YuAmwq8o3NzmTBNrUqZ9meiNwF8Xzi9z90/EZdwY50302AmiwwunN+ghkzJWSibXuusaK2U5Y0vp6KkT4TPxnyG8SbLmeMo0sSZl2peJufKxuzsUTt8d/94JbNrhfzcnvGGZmkkZqzKTa901jlW5nKqpMqBJk6ZmTYkN5DDC7/79GeHb/NYD9orzDu8hU0vzzLXuGseqXE7l/V7zSrQFcAHwC+BzwKTCZZe3OdPEmpRpX6aXBz4wC7gBeBp4Cvg5sH/pMTtmJmWsHurJte5VHiv1eg2yWV8LHAXsRDje8Gbil28Td6i3NdPEmpRpX6aHBrLFGI/Td6RmUptMSibHuusaK3U5Y011N+s7SucPI7xJsiWjH69sZaaJNSnTvkycn9LQ7gNmdLj8CODXPWTqbJ7Z1V3jWJXLqZoqA71MccVaqzRvX8KhLE+0OdPEmpRpXybOS2kgBxB+C3DrwuXHAouAaT1kammeudZd41iVy6ma6m7WnwL26jB/JnBtmzNNrEmZ9mXi+aQHPvA2QqPfATiZsFtlw9Kyx8ykjNVDPbnWvcpjpV6vsaZam7UmTZoGM6U+8IG3EPahXkFpqz01U0fzbEHdqzxW6vXqep/3YSXaDziS0ssC4Ii2Z5pYkzLtyxTmdX3gA8uA5+Pf/yJ858jI+edTMylj9ZLJre66xuplOV1vt5RQ6gScQHhH9GTg18DHC5ctaHOmiTUp075MXQ/8Hh7TtTXPXOuuY6xarlPNN9AiwscqIXyE8krgpHh+YZszTaxJmfZlUifCd2AfTfgBg9kjy+01M+ipaXXXNVYdy1mDek1095cA3P1Z4H8Ck83sYmDNlmeaWJMy7ctgZmuZ2dFmdqqZzY5fy1l2NjBMeAI4APjWymRSxkqsJ9e6axkr8XqNreZnoR/T+d3s44FX2pxpYk3KtC8Tz18InEv48qDLgVM6/M+iwumJFHaj9JhJGasyk2vdNY5VuZyqqadw5cLizxB1uWzzNmeaWJMy7cvE0ykNZMFY53vI1NI8c627xrEql1M11foVqcDGwLPA7wHMbG/g3cCjhH01bc40sSZl2pcBeHHkhLu/ZGZ0sKOZPR9PG7B2PG/h33xyYiZlrJRMrnXXNVbKcsZU9z7ri4B1AMxsJ+Bi4DFgR+C0lmeaWJMy7ctAfODHaRnwpyOnRxqCu09w98lxWs/dJxZOT07NpIyVmMmy7rrGSlzO2HrdFB9rYvmvQ/wn4Bvx9BqMflVgKzNNrEmZ9mU0jd+p7i3r4vb/PsDPANz9lcJlbc00sSZl2peRcarufdbXmdlFwBPAhsB1AGa2GfCfLc80sSZl2peRcaruZn008H5gM2BPH/3J962AKS3PNLEmZdqXkfGqX/tXCN8U9k3gEWAuhY/Otj3TxJqUaV9G0/iaat2yNrNtgEPj9BThYHFz973bnmliTcq0LyPjWJ2dH3iF8HM2WxXmPTQeMk2sSZn2ZTSN36nuo0HeQ3hzZK6Z/YuZvY0V38Vua6aJNSnTvoyMV/14BiAc2P9B4EeErws8HXj7eMg0sSZl2pfRNP6m/g8QDkGaDfxsvGWaWJMy7ctoGh+TuTsiItJsde+zFhGRPlCzFhHJgJq1iEgG1KxFRDKgZi0ikoH/BqEetwFrzQX/AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "count_df['all_variants'].plot.bar()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "226cc535",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "applying filters\n",
      "removing variants outside of peak interval\n",
      "removed unkwon genotypes (e.g \"./.\")\n",
      "replaced phased genotype (e.g \"0|1\" -> \"0/1\")\n",
      "done\n",
      "getting table\n"
     ]
    }
   ],
   "source": [
    "dff = gdf.get_table()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4fa05933",
   "metadata": {},
   "outputs": [],
   "source": [
    "interval_df = pd.read_csv('data/peaks_with_overlaps1mb.csv', index_col=0)\n",
    "# interval_df.CHROM = interval_df.CHROM.str.replace('chr','')\n",
    "interval_df[['CHROM','FROM','TO']].to_csv('data/peak_bcftools_format.tsv', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ee8b9ffe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr1',\n",
       " 'chr10',\n",
       " 'chr11',\n",
       " 'chr12',\n",
       " 'chr13',\n",
       " 'chr14',\n",
       " 'chr14_GL000225v1_random',\n",
       " 'chr14_KI270722v1_random',\n",
       " 'chr14_KI270726v1_random',\n",
       " 'chr15',\n",
       " 'chr15_KI270727v1_random',\n",
       " 'chr16',\n",
       " 'chr16_KI270728v1_random',\n",
       " 'chr17',\n",
       " 'chr18',\n",
       " 'chr19',\n",
       " 'chr2',\n",
       " 'chr20',\n",
       " 'chr21',\n",
       " 'chr22',\n",
       " 'chr3',\n",
       " 'chr4',\n",
       " 'chr5',\n",
       " 'chr6',\n",
       " 'chr7',\n",
       " 'chr8',\n",
       " 'chr9',\n",
       " 'chrUn_GL000219v1',\n",
       " 'chrUn_KI270752v1',\n",
       " 'chrX',\n",
       " 'chrY']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chr2 chr20 chr21 chr22 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chrX chrY"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "dev-venv",
   "language": "python",
   "name": "dev-venv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
